<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Roey Angel">
<meta name="dcterms.date" content="2024-06-17">

<title>Millipedes SIP</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="37_SIP_diff_abund_between_DESeq2_files/libs/clipboard/clipboard.min.js"></script>
<script src="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/quarto.js"></script>
<script src="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/popper.min.js"></script>
<script src="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/anchor.min.js"></script>
<link href="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="37_SIP_diff_abund_between_DESeq2_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="37_SIP_diff_abund_between_DESeq2_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="37_SIP_diff_abund_between_DESeq2_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="37_SIP_diff_abund_between_DESeq2_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="37_SIP_diff_abund_between_DESeq2_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="37_SIP_diff_abund_between_DESeq2_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Millipedes SIP</h1>
<p class="subtitle lead">05 SIP differential abundance modelling: comparing between gradients using DESeq2</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Roey Angel </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 17, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/style settings_4b3056e7d7040298dd6f8c60487f74e8">

</div>
<section id="differential-abundance-modelling-of-sip-gradients" class="level2">
<h2 class="anchored" data-anchor-id="differential-abundance-modelling-of-sip-gradients">Differential abundance modelling of SIP gradients</h2>
<p>Here we attempt to detect ASVs that were labelled with <sup>13</sup>C using differential abundance modelling. Using DESeq2 <span class="citation" data-cites="love_moderated_2014">(<a href="#ref-love_moderated_2014" role="doc-biblioref">Love, Huber and Anders 2014</a>)</span> we compare the relative abundance of each ASV in the fractions where <sup>13</sup>C-labelled RNA is expected to be found (&gt;1.795 g ml<sup>-1</sup>; AKA ‘heavy’ fractions) to the fractions where unlabelled RNA is expected to be found (&lt;1.795 g ml<sup>-1</sup>; AKA ‘light’ fractions). The method has been previously described in Angel et al., <span class="citation" data-cites="angel_application_2018">(<a href="#ref-angel_application_2018" role="doc-biblioref"><strong>angel_application_2018?</strong></a>)</span>.</p>
<section id="setting-general-parameters" class="level3">
<h3 class="anchored" data-anchor-id="setting-general-parameters">Setting general parameters:</h3>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/general parameters_fc6582442ac5f0f6007a1a810416af67">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2021</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>alpha_thresh <span class="ot">&lt;-</span> <span class="fl">0.1</span> <span class="co"># only keep ASVs with a diff abundance p-value below this</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>LFC_thresh <span class="ot">&lt;-</span> <span class="dv">0</span>  <span class="co"># only keep ASVs with a diff abundance LFC above this</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>prev_thresh <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="co"># only keep ASVs that appear in at least X samples (fractions)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>rep_thresh <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="co"># only keep ASVs that appear significant in at least X replicates</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>abund_thresh <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>rare_thresh <span class="ot">&lt;-</span> <span class="fl">0.0045</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>vars2test <span class="ot">&lt;-</span> <span class="st">"Label..13C."</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>ref_level <span class="ot">&lt;-</span> <span class="st">"Unlabelled"</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">"./37_DADA2_pseudo/"</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>Metadata_table <span class="ot">&lt;-</span> <span class="st">"./37_Millipedes_SIP_metadata.csv"</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Seq_table &lt;- "DADA2.seqtab_nochim_decontam.tsv"</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Seq_file &lt;- "DADA2.Seqs_decontam.fa"</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>Proj_name <span class="ot">&lt;-</span> <span class="st">"Millipedes_SIP"</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>Ps_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(Proj_name, <span class="st">"_seq_prev_filt.Rds"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Tree_file &lt;- "./Tree/DADA2.Seqs_decontam_filtered.filtered.align.treefile"</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>Tree_file <span class="ot">&lt;-</span> <span class="st">"./Tree/DADA2_reps_seq_prev_filt.filtered.align.treefile"</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>Var1 <span class="ot">=</span> <span class="st">"Replicate"</span> <span class="co"># e.g sampling point / replicate</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>Var2 <span class="ot">=</span> <span class="st">"Day"</span> <span class="co"># e.g. a treatment or a manipulation</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>Var3 <span class="ot">=</span> <span class="st">"Species"</span> <span class="co"># e.g. a treatment/manipulation or an important covariant</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>Var4 <span class="ot">=</span> <span class="st">"Label..13C."</span> <span class="co"># e.g. an important covariant</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-phyloseq-object" class="level3">
<h3 class="anchored" data-anchor-id="read-phyloseq-object">Read phyloseq object</h3>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/load data_7e6312d71532f5f3f623f2b0bf77dd49">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load phylogenetic tree</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>Tree <span class="ot">&lt;-</span> <span class="fu">read_tree</span>(<span class="fu">paste0</span>(data_path, Tree_file))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load and merge  phyloseq object</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">readRDS</span>(<span class="fu">paste0</span>(data_path, Ps_file)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_phyloseq</span>(.,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">phy_tree</span>(Tree)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="ot">-&gt;</span> Ps_obj_SIP </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add 18S data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">read_delim</span>(Metadata_table,</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">delim =</span> <span class="st">","</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"merged_sample_name"</span>, <span class="st">`</span><span class="at">18S_copies (undiluted RNA)</span><span class="st">`</span>) <span class="ot">-&gt;</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  fungi_quant</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 337 Columns: 27</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (10): sample_ID, merged_sample_name, Read1_file, Long_name, Species, Col...
dbl (16): Day, Replicate, TNA ext. batch, RNA (ng/ul), Volume per gradient (...
lgl  (1): Control

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save filtered phyloseq object for PICRUSt</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Ps_obj_Labelled &lt;- subset_samples(Ps_obj_SIP, Label..13C. == 'Labelled')</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#seqs &lt;- refseq(Ps_obj_Labelled)</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#otab &lt;- otu_table(Ps_obj_Labelled)</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(seqs, "Ps_obj_Labelled_seqs.tsv", quote=FALSE)</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(otab, "Ps_obj_Labelled_table.tsv", quote=FALSE,sep="\t")</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Then, run the seqs file through this one liner to clean it up and create a fasta file.</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># grep -v '^x' Ps_obj_Labelled_seqs.tsv | awk '{print "&gt;"$1"\n"$2}' &gt; Ps_obj_Labelled_seqs.fa; rm Ps_obj_Labelled_seqs.tsv</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="fu">get_variable</span>(Ps_obj_SIP) <span class="sc">%&gt;%</span>  </span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., fungi_quant, <span class="at">by =</span> <span class="st">"merged_sample_name"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">`</span><span class="at">18S_copies (undiluted RNA)</span><span class="st">`</span>) <span class="ot">-&gt;</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>FungiQuant</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Label..13C., <span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Density.zone))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Day <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Day, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"3"</span>, <span class="st">"7"</span>, <span class="st">"14"</span>, <span class="st">"21"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Label..13C. <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Label..13C., <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Labelled"</span>, <span class="st">"Unlabelled"</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Species, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Epibolus pulchripes"</span>, <span class="st">"Glomeris connexa"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Fraction.no. <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Fraction.no., <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Density.zone <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(<span class="fu">sample_data</span>(Ps_obj_SIP)<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span> <span class="sc">&gt;</span> <span class="fl">1.8</span>, <span class="st">"Heavy"</span>, <span class="st">"Light"</span>), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Light"</span>, <span class="st">"Heavy"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_SIP <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_variable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="sc">!!</span><span class="fu">sym</span>(Var3), <span class="sc">!!</span><span class="fu">sym</span>(Var2), <span class="sc">!!</span><span class="fu">sym</span>(Var4), <span class="sc">!!</span><span class="fu">sym</span>(Var1)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">qPCR_fraction =</span> X16S_copies..undiluted.RNA.<span class="sc">/</span><span class="fu">sum</span>(X16S_copies..undiluted.RNA.) <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>) <span class="sc">%&gt;%</span> <span class="co"># times 10^6 is important to not end up with very small fractions! </span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fungi_quant_fraction =</span> FungiQuant<span class="sc">/</span><span class="fu">sum</span>(FungiQuant) <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>) <span class="ot">-&gt;</span> </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_data</span>(Ps_obj_SIP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Assuming first column, `sample_ID`, contains the sample names
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="ot">&lt;-</span> Ps_obj_SIP</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">otu_table</span>(Ps_obj_abs) <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(., <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">"matrix"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2_dfc</span>(., <span class="fu">get_variable</span>(Ps_obj_abs, <span class="st">"qPCR_fraction"</span>) , <span class="st">`</span><span class="at">*</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(., <span class="at">row.names =</span> <span class="fu">taxa_names</span>(Ps_obj_abs)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">otu_table</span>(., <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mark rare taxa</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&lt;&gt;%</span> <span class="fu">mark_rare_taxa</span>(., <span class="at">rank =</span> <span class="st">"Phylum"</span>, <span class="at">rare_thresh =</span> rare_thresh)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `across(rank, ~if_else(Phylum %in% Rare_phyla, "Rare",
  Phylum))`.
Caused by warning:
! Using an external vector in selections was deprecated in tidyselect 1.1.0.
ℹ Please use `all_of()` or `any_of()` instead.
  # Was:
  data %&gt;% select(rank)

  # Now:
  data %&gt;% select(all_of(rank))

See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
</div>
</section>
<section id="plot-the-sip-gradients" class="level3">
<h3 class="anchored" data-anchor-id="plot-the-sip-gradients">Plot the SIP gradients</h3>
<p>First I’ll plot the 16S copy numbers vs density</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/plot SIP fractions_270a7f0357ed20d0fe6ec9e534a2e2dc">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_variable</span>(Ps_obj_abs) <span class="sc">%&gt;%</span> </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">qPCR_fraction =</span> qPCR_fraction<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">4</span>,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">fungi_quant_fraction =</span> fungi_quant_fraction<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">4</span>) <span class="sc">%&gt;%</span> <span class="co"># bring it back from ppm to %</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rep_label =</span> <span class="fu">paste</span>(Replicate, Label..13C.)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rep_label =</span> <span class="fu">factor</span>(dplyr<span class="sc">::</span><span class="fu">recode</span>(Rep_label, </span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">1 Labelled</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"1"</span>, </span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">2 Labelled</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"2"</span>, </span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">3 Labelled</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"3"</span>,</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">1 Unlabelled</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Unlabelled"</span>),</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>                            <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Unlabelled"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">recode_factor</span>(Species, <span class="st">`</span><span class="at">Epibolus pulchripes</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"*E. pulchripes*"</span>, <span class="st">`</span><span class="at">Glomeris connexa</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"*G. connexa*"</span>)) <span class="ot">-&gt;</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  qPCR_df</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_SIP_fractions</span>(qPCR_df, </span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> <span class="st">"Rep_label"</span>, </span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xticks =</span> <span class="fu">seq</span>(<span class="fl">1.71</span>, <span class="fl">1.85</span>, <span class="at">by =</span> <span class="fl">0.03</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/plot%20SIP%20fractions-1.png" class="img-fluid" width="960"></p>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_SIP_fractions</span>(qPCR_df, </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="st">"fungi_quant_fraction"</span>, </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> <span class="st">"Rep_label"</span>, </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xticks =</span> <span class="fu">seq</span>(<span class="fl">1.71</span>, <span class="fl">1.85</span>, <span class="at">by =</span> <span class="fl">0.03</span>)) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Density (g ml&lt;sup&gt;-1&lt;/sup&gt;)"</span>, <span class="at">y =</span> <span class="st">"18S rRNA fungal copies (%)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/plot%20SIP%20fractions-2.png" class="img-fluid" width="960"></p>
</div>
</div>
</section>
<section id="beta-diversity-join-analysis" class="level3">
<h3 class="anchored" data-anchor-id="beta-diversity-join-analysis">Beta Diversity Join Analysis</h3>
<p>Let us look first at the dissimilarity in community composition between the different fractions. If the labelling was strong enough we should see a deviation of (some of) the heavy fractions from the light ones. However, a lack of a significant deviation does not mean unsuccessful labelling because if only a small minority of the community was labelled we might not see it here (but we will, hopefully, see it using DESeq2 modelling).</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/beta div joint_bed3b108a76c1e0eab4d213b1f242748">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join analysis</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lib_dist</span>(Ps_obj_abs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20joint-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>(mod0 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_abs), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Lib.size,</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_abs),</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_abs), method = "horn") ~ Lib.size, data = get_variable(Ps_obj_abs), permutations = 999)
          Df SumOfSqs      R2      F Pr(&gt;F)    
Lib.size   1    7.538 0.06726 22.859  0.001 ***
Residual 317  104.541 0.93274                  
Total    318  112.079 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>(mod3 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_abs), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Species <span class="sc">*</span> Day <span class="sc">+</span> Species<span class="sc">:</span>Density.zone,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_abs),</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_abs), method = "horn") ~ Species * Day + Species:Density.zone, data = get_variable(Ps_obj_abs), permutations = 999)
                      Df SumOfSqs      R2       F Pr(&gt;F)    
Species                1   28.610 0.25527 142.824  0.001 ***
Day                    3   10.206 0.09106  16.983  0.001 ***
Species:Day            3    8.060 0.07191  13.412  0.001 ***
Species:Density.zone   2    3.304 0.02948   8.248  0.001 ***
Residual             309   61.899 0.55227                   
Total                318  112.079 1.00000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>Species_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_abs), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_abs, <span class="st">"Species"</span>))</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">permutest</span>(Species_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
           Df  Sum Sq Mean Sq      F N.Perm Pr(&gt;F)    
Groups      1  0.5582 0.55825 17.433    999  0.001 ***
Residuals 317 10.1510 0.03202                         
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Species_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20joint-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>Day_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_abs), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_abs, <span class="st">"Day"</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">permutest</span>(Day_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
           Df Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
Groups      3 0.0373 0.012429 1.0283    999  0.417
Residuals 315 3.8073 0.012087                     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Day_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20joint-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>Density_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_abs), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_abs, <span class="st">"Density.zone"</span>))</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">permutest</span>(Density_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
           Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)  
Groups      1 0.03762 0.037623 4.7801    999  0.033 *
Residuals 317 2.49503 0.007871                       
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Density_disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20joint-4.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/ordinations_b6060c7c57a6dfa5fe70c7f4dd42684c">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>Ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(Ps_obj_abs, <span class="st">"CAP"</span>, <span class="st">"horn"</span>, </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">formula =</span>  <span class="sc">~</span> Species <span class="sc">*</span> Day <span class="sc">+</span> Species<span class="sc">:</span>Density.zone)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>explained <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">eigenvals</span>(Ord)<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">eigenvals</span>(Ord)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="at">nsmall =</span> <span class="dv">1</span>))</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>Ord_plt <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(Ps_obj_abs, Ord, <span class="at">type =</span> <span class="st">"Sample.no."</span>, <span class="at">color =</span> <span class="st">"Label..13C."</span>, <span class="at">justDF =</span> <span class="cn">TRUE</span>)</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>p_ord_joint <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Ord_plt) <span class="sc">+</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> CAP1,</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> CAP2,</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> Label..13C.,</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> Density..g.ml.<span class="fl">1.</span>,</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">shape =</span> Species</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>             ), <span class="at">alpha =</span> <span class="dv">2</span> <span class="sc">/</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Labelling"</span>), </span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">size =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Density (g ml&lt;sup&gt;-1&lt;/sup&gt;)"</span>),</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">shape =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Species"</span>)) <span class="sc">+</span></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_locuszoom</span>() <span class="sc">+</span> </span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">sprintf</span>(<span class="st">"CAP1 (%s%%)"</span>, explained[<span class="dv">1</span>]),</span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">sprintf</span>(<span class="st">"CAP2 (%s%%)"</span>, explained[<span class="dv">2</span>])) <span class="sc">+</span></span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="fu">sqrt</span>(explained[<span class="dv">2</span>] <span class="sc">/</span> explained[<span class="dv">1</span>])) <span class="sc">+</span></span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">legend.justification =</span> <span class="st">"top"</span>,</span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.title =</span> <span class="fu">element_markdown</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.direction=</span><span class="st">'vertical'</span>,</span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>)</span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">+</span></span>
<span id="cb59-27"><a href="#cb59-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">breaks =</span> <span class="fu">round</span>(<span class="fu">c</span>(<span class="fu">seq</span>(<span class="fu">min</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb59-28"><a href="#cb59-28" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">max</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb59-29"><a href="#cb59-29" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">length.out =</span> <span class="dv">4</span>), </span>
<span id="cb59-30"><a href="#cb59-30" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">1</span>), <span class="dv">4</span>),</span>
<span id="cb59-31"><a href="#cb59-31" aria-hidden="true" tabindex="-1"></a>                        <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb59-32"><a href="#cb59-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">guides</span>(<span class="at">size =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="fu">seq</span>(<span class="dv">2</span>, <span class="dv">5</span>)))) <span class="sc">+</span></span>
<span id="cb59-33"><a href="#cb59-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Species <span class="sc">~</span> Day) <span class="sc">+</span></span>
<span id="cb59-34"><a href="#cb59-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb59-35"><a href="#cb59-35" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb59-36"><a href="#cb59-36" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_ord_joint)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/ordinations-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save_figure(paste0(fig.path, "Oridnation"),</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="co">#             p_ord_joint,</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="co">#             pwidth = 10,</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a><span class="co">#             pheight = 8,</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="co">#             dpi = 600)</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co"># knitr::include_graphics(paste0(fig.path, "Oridnation", ".png"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="beta-diversity-analysis-for-epibolus-pulchripes" class="level3">
<h3 class="anchored" data-anchor-id="beta-diversity-analysis-for-epibolus-pulchripes">Beta diversity analysis for Epibolus pulchripes</h3>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/beta div for Epibolus pulchripes_38b77ddb1336802654327f9840448950">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_Epi <span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(Ps_obj_abs, Species <span class="sc">==</span> <span class="st">'Epibolus pulchripes'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lib_dist</span>(Ps_obj_Epi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Epibolus%20pulchripes-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>(mod0 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Epi), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Lib.size,</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Epi),</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Epi), method = "horn") ~ Lib.size, data = get_variable(Ps_obj_Epi), permutations = 999)
          Df SumOfSqs      R2      F Pr(&gt;F)    
Lib.size   1    1.046 0.02881 4.6577  0.001 ***
Residual 157   35.251 0.97119                  
Total    158   36.297 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>(mod1 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Epi), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Day <span class="sc">*</span> Density.zone,</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Epi),</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Epi), method = "horn") ~ Day * Density.zone, data = get_variable(Ps_obj_Epi), permutations = 999)
                  Df SumOfSqs      R2       F Pr(&gt;F)    
Day                3    9.987 0.27516 21.2666  0.001 ***
Density.zone       1    2.083 0.05738 13.3047  0.001 ***
Day:Density.zone   3    0.589 0.01623  1.2544  0.195    
Residual         151   23.638 0.65123                   
Total            158   36.297 1.00000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>(mod2 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Epi), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Day <span class="sc">+</span> Density.zone,</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Epi),</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Epi), method = "horn") ~ Day + Density.zone, data = get_variable(Ps_obj_Epi), permutations = 999)
              Df SumOfSqs      R2      F Pr(&gt;F)    
Day            3    9.987 0.27516 21.162  0.001 ***
Density.zone   1    2.083 0.05738 13.239  0.001 ***
Residual     154   24.227 0.66746                  
Total        158   36.297 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>mod2 <span class="sc">%&gt;%</span> </span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df SumOfSqs</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">F</th>
<th style="text-align: right;">Pr(&gt;F)</th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Day</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">9.987</td>
<td style="text-align: right;">0.275</td>
<td>21.162</td>
<td>0.001</td>
</tr>
<tr class="even">
<td>Density.zone</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2.083</td>
<td style="text-align: right;">0.057</td>
<td>13.239</td>
<td>0.001</td>
</tr>
<tr class="odd">
<td>Residual</td>
<td style="text-align: right;">154</td>
<td style="text-align: right;">24.227</td>
<td style="text-align: right;">0.667</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Total</td>
<td style="text-align: right;">158</td>
<td style="text-align: right;">36.297</td>
<td style="text-align: right;">1.000</td>
<td>NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>Day_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Epi), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_Epi, <span class="st">"Day"</span>))</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(Day_disp) <span class="sc">%&gt;%</span> </span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df Sum Sq</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Sq F v</th>
<th>alue Pr(</th>
<th style="text-align: left;">&gt;F)</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Groups</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.830</td>
<td>0.277</td>
<td style="text-align: left;">8.482</td>
<td>0</td>
</tr>
<tr class="even">
<td>Residuals</td>
<td style="text-align: right;">155</td>
<td style="text-align: right;">5.055</td>
<td>0.033</td>
<td style="text-align: left;">NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Day_disp, <span class="at">main =</span> <span class="st">"Day BetaDisper in Epibolus Hindgut Samples"</span>, <span class="at">cex.lab =</span> <span class="fl">1.3</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Epibolus%20pulchripes-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>Density_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Epi), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_Epi, <span class="st">"Density.zone"</span>))</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(Density_disp) <span class="sc">%&gt;%</span> </span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df Sum Sq</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Sq F v</th>
<th>alue Pr(</th>
<th style="text-align: left;">&gt;F)</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Groups</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.337</td>
<td>0.337</td>
<td style="text-align: left;">9.466</td>
<td>0.002</td>
</tr>
<tr class="even">
<td>Residuals</td>
<td style="text-align: right;">157</td>
<td style="text-align: right;">5.588</td>
<td>0.036</td>
<td style="text-align: left;">NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Density_disp, <span class="at">main =</span> <span class="st">"Density BetaDisper in Epibolus Hindgut Samples"</span>, <span class="at">cex.lab =</span> <span class="fl">1.3</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Epibolus%20pulchripes-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/ordinations for Epibolus pulchripes_d1753ddabbb1c7297d543cd92f753c8e">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>Ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(Ps_obj_Epi, <span class="st">"CAP"</span>, <span class="st">"horn"</span>, </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">formula =</span>  <span class="sc">~</span> Day <span class="sc">+</span> Density.zone)</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>explained <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">eigenvals</span>(Ord)<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">eigenvals</span>(Ord)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="at">nsmall =</span> <span class="dv">1</span>))</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>Ord_plt <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(Ps_obj_Epi, Ord, <span class="at">type =</span> <span class="st">"Sample.no."</span>, <span class="at">color =</span> <span class="st">"Label..13C."</span>, <span class="at">justDF =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>p_ord_EP <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Ord_plt) <span class="sc">+</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> CAP1,</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> CAP2,</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> Label..13C.,</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> Density..g.ml.<span class="fl">1.</span></span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>             ), <span class="at">alpha =</span> <span class="dv">2</span> <span class="sc">/</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Labelling"</span>), </span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">size =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Density (g ml&lt;sup&gt;-1&lt;/sup&gt;)"</span>),</span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">shape =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Species"</span>)) <span class="sc">+</span></span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_locuszoom</span>() <span class="sc">+</span> </span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">sprintf</span>(<span class="st">"CAP1 (%s%%)"</span>, explained[<span class="dv">1</span>]),</span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">sprintf</span>(<span class="st">"CAP2 (%s%%)"</span>, explained[<span class="dv">2</span>])) <span class="sc">+</span></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_fixed(ratio = sqrt(explained[2] / explained[1])) +</span></span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="co">#legend.justification = "top",</span></span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a>         <span class="co"># legend.title = element_markdown(size = 11),</span></span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>         <span class="co"># legend.direction='vertical',</span></span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>)</span>
<span id="cb83-25"><a href="#cb83-25" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">+</span></span>
<span id="cb83-26"><a href="#cb83-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">breaks =</span> <span class="fu">round</span>(<span class="fu">c</span>(<span class="fu">seq</span>(<span class="fu">min</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb83-27"><a href="#cb83-27" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">max</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb83-28"><a href="#cb83-28" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">length.out =</span> <span class="dv">4</span>), </span>
<span id="cb83-29"><a href="#cb83-29" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">1</span>), <span class="dv">4</span>),</span>
<span id="cb83-30"><a href="#cb83-30" aria-hidden="true" tabindex="-1"></a>                        <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb83-31"><a href="#cb83-31" aria-hidden="true" tabindex="-1"></a>   <span class="co"># guides(size = guide_legend(override.aes = list(size = seq(2, 5)))) +</span></span>
<span id="cb83-32"><a href="#cb83-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">round</span>(<span class="fu">seq</span>(<span class="fu">min</span>(Ord_plt<span class="sc">$</span>CAP1), <span class="fu">max</span>(Ord_plt<span class="sc">$</span>CAP1), <span class="at">by =</span> <span class="dv">1</span>),<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb83-33"><a href="#cb83-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Species <span class="sc">~</span> Day) <span class="sc">+</span></span>
<span id="cb83-34"><a href="#cb83-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb83-35"><a href="#cb83-35" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb83-36"><a href="#cb83-36" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_ord_EP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/ordinations%20for%20Epibolus%20pulchripes-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save_figure(paste0(fig.path, "Oridnation for Epibolus pulchripes"), </span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co">#             p_ord_joint, </span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="co">#             pwidth = 10, </span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="co">#             pheight = 8,</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="co">#             dpi = 600)</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="co"># knitr::include_graphics(paste0(fig.path, "Oridnation for Epibolus pulchripes", ".png"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="beta-diversity-analysis-for-glomeris-connexa" class="level3">
<h3 class="anchored" data-anchor-id="beta-diversity-analysis-for-glomeris-connexa">Beta diversity analysis for Glomeris connexa</h3>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/beta div for Glomeris connexa_b92fe679d382b0aedb66a7e888b50777">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_Glo <span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(Ps_obj_abs, Species <span class="sc">==</span> <span class="st">'Glomeris connexa'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lib_dist</span>(Ps_obj_Glo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Glomeris%20connexa-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>(mod0 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Glo), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Lib.size,</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Glo),</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Glo), method = "horn") ~ Lib.size, data = get_variable(Ps_obj_Glo), permutations = 999)
          Df SumOfSqs      R2      F Pr(&gt;F)    
Lib.size   1    1.444 0.03062 4.9901  0.001 ***
Residual 158   45.728 0.96938                  
Total    159   47.173 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>(mod1 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Glo), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Day <span class="sc">*</span> Density.zone,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Glo),</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Glo), method = "horn") ~ Day * Density.zone, data = get_variable(Ps_obj_Glo), permutations = 999)
                  Df SumOfSqs      R2       F Pr(&gt;F)    
Day                3    8.279 0.17550 11.4582  0.001 ***
Density.zone       1    1.222 0.02590  5.0727  0.001 ***
Day:Density.zone   3    1.063 0.02254  1.4716  0.085 .  
Residual         152   36.609 0.77606                   
Total            159   47.173 1.00000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>(mod2 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Glo), <span class="at">method =</span> <span class="st">"horn"</span>) <span class="sc">~</span> Day <span class="sc">+</span> Density.zone,</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">get_variable</span>(Ps_obj_Glo),</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegdist(otu_table(Ps_obj_Glo), method = "horn") ~ Day + Density.zone, data = get_variable(Ps_obj_Glo), permutations = 999)
              Df SumOfSqs     R2       F Pr(&gt;F)    
Day            3    8.279 0.1755 11.3545  0.001 ***
Density.zone   1    1.222 0.0259  5.0268  0.001 ***
Residual     155   37.672 0.7986                   
Total        159   47.173 1.0000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>mod2 <span class="sc">%&gt;%</span> </span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>, <span class="st">"condensed"</span>, <span class="st">"responsive"</span>), <span class="at">full_width =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in kable_styling(., bootstrap_options = c("striped", "hover",
"condensed", : Please specify format in kable. kableExtra can customize either
HTML or LaTeX outputs. See https://haozhu233.github.io/kableExtra/ for details.</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df SumOfSqs</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">F</th>
<th style="text-align: right;">Pr(&gt;F)</th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Day</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">8.279</td>
<td style="text-align: right;">0.176</td>
<td>11.355</td>
<td>0.001</td>
</tr>
<tr class="even">
<td>Density.zone</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.222</td>
<td style="text-align: right;">0.026</td>
<td>5.027</td>
<td>0.001</td>
</tr>
<tr class="odd">
<td>Residual</td>
<td style="text-align: right;">155</td>
<td style="text-align: right;">37.672</td>
<td style="text-align: right;">0.799</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Total</td>
<td style="text-align: right;">159</td>
<td style="text-align: right;">47.173</td>
<td style="text-align: right;">1.000</td>
<td>NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>Day_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Glo), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_Glo, <span class="st">"Day"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in betadisper(vegdist(otu_table(Ps_obj_Glo), method = "horn"),
get_variable(Ps_obj_Glo, : some squared distances are negative and changed to
zero</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(Day_disp) <span class="sc">%&gt;%</span> </span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df Sum Sq</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Sq F v</th>
<th>alue Pr(</th>
<th style="text-align: left;">&gt;F)</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Groups</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.387</td>
<td>0.462</td>
<td style="text-align: left;">15.239</td>
<td>0</td>
</tr>
<tr class="even">
<td>Residuals</td>
<td style="text-align: right;">156</td>
<td style="text-align: right;">4.732</td>
<td>0.030</td>
<td style="text-align: left;">NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Day_disp, <span class="at">main =</span> <span class="st">"Day BetaDisper in Glomeris Hindgut Samples"</span>, <span class="at">cex.lab =</span> <span class="fl">1.3</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Glomeris%20connexa-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>Density_disp <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(<span class="fu">vegdist</span>(<span class="fu">otu_table</span>(Ps_obj_Glo), <span class="at">method =</span> <span class="st">"horn"</span>), <span class="fu">get_variable</span>(Ps_obj_Glo, <span class="st">"Density.zone"</span>))</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(Density_disp) <span class="sc">%&gt;%</span> </span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">format =</span> <span class="st">"pandoc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th>Df Sum Sq</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Sq F v</th>
<th>alue Pr(</th>
<th style="text-align: left;">&gt;F)</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Groups</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.063</td>
<td>0.063</td>
<td style="text-align: left;">2.818</td>
<td>0.095</td>
</tr>
<tr class="even">
<td>Residuals</td>
<td style="text-align: right;">158</td>
<td style="text-align: right;">3.557</td>
<td>0.023</td>
<td style="text-align: left;">NA</td>
<td>NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Density_disp, <span class="at">main =</span> <span class="st">"Density BetaDisper in Glomeris Hindgut Samples"</span>, <span class="at">cex.lab =</span> <span class="fl">1.3</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/beta%20div%20for%20Glomeris%20connexa-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/ordinations for Glomeris connexa_593646b95a64c3df501ebb2f2948c245">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>Ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(Ps_obj_Glo, <span class="st">"CAP"</span>, <span class="st">"horn"</span>, </span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">formula =</span>  <span class="sc">~</span> Day <span class="sc">+</span> Density.zone)</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>explained <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">eigenvals</span>(Ord)<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">eigenvals</span>(Ord)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="at">nsmall =</span> <span class="dv">1</span>))</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>Ord_plt <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(Ps_obj_Glo, Ord, <span class="at">type =</span> <span class="st">"Sample.no."</span>, <span class="at">color =</span> <span class="st">"Label..13C."</span>, <span class="at">justDF =</span> <span class="cn">TRUE</span>)</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>p_ord_GC <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Ord_plt) <span class="sc">+</span></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> CAP1,</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> CAP2,</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> Label..13C.,</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> Density..g.ml.<span class="fl">1.</span></span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>             ), <span class="at">alpha =</span> <span class="dv">2</span> <span class="sc">/</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Labelling"</span>), </span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">size =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Density (g ml&lt;sup&gt;-1&lt;/sup&gt;)"</span>),</span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">shape =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"Species"</span>)) <span class="sc">+</span></span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_locuszoom</span>() <span class="sc">+</span> </span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">sprintf</span>(<span class="st">"CAP1 (%s%%)"</span>, explained[<span class="dv">1</span>]),</span>
<span id="cb110-18"><a href="#cb110-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">sprintf</span>(<span class="st">"CAP2 (%s%%)"</span>, explained[<span class="dv">2</span>])) <span class="sc">+</span></span>
<span id="cb110-19"><a href="#cb110-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># coord_fixed(ratio = sqrt(explained[2] / explained[1])) +</span></span>
<span id="cb110-20"><a href="#cb110-20" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme</span>(<span class="at">legend.justification =</span> <span class="st">"top"</span>,</span>
<span id="cb110-21"><a href="#cb110-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.title =</span> <span class="fu">element_markdown</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb110-22"><a href="#cb110-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.direction=</span><span class="st">'vertical'</span>,</span>
<span id="cb110-23"><a href="#cb110-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb110-24"><a href="#cb110-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>)</span>
<span id="cb110-25"><a href="#cb110-25" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">+</span></span>
<span id="cb110-26"><a href="#cb110-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">breaks =</span> <span class="fu">round</span>(<span class="fu">c</span>(<span class="fu">seq</span>(<span class="fu">min</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb110-27"><a href="#cb110-27" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">max</span>(Ord_plt<span class="sc">$</span>Density..g.ml.<span class="fl">1.</span>), </span>
<span id="cb110-28"><a href="#cb110-28" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">length.out =</span> <span class="dv">4</span>), </span>
<span id="cb110-29"><a href="#cb110-29" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">1</span>), <span class="dv">4</span>),</span>
<span id="cb110-30"><a href="#cb110-30" aria-hidden="true" tabindex="-1"></a>                        <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb110-31"><a href="#cb110-31" aria-hidden="true" tabindex="-1"></a>   <span class="co"># guides(size = guide_legend(override.aes = list(size = seq(2, 5)))) +</span></span>
<span id="cb110-32"><a href="#cb110-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">round</span>(<span class="fu">seq</span>(<span class="fu">min</span>(Ord_plt<span class="sc">$</span>CAP1), <span class="fu">max</span>(Ord_plt<span class="sc">$</span>CAP1), <span class="at">by =</span> <span class="dv">1</span>),<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb110-33"><a href="#cb110-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Species <span class="sc">~</span> Day) <span class="sc">+</span></span>
<span id="cb110-34"><a href="#cb110-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb110-35"><a href="#cb110-35" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb110-36"><a href="#cb110-36" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_ord_GC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/ordinations%20for%20Glomeris%20connexa-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>patched_ord <span class="ot">&lt;-</span> (p_ord_EP <span class="sc">+</span> p_ord_GC) <span class="sc">+</span> </span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>) </span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Patched_ordination"</span>),</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>            patched_ord,</span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">pwidth =</span> <span class="dv">10</span>,</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">pheight =</span> <span class="dv">8</span>,</span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a><span class="co"># knitr::include_graphics(paste0(fig.path, "Oridnation for Glomeris connexa", ".png"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="subset-the-dataset" class="level3">
<h3 class="anchored" data-anchor-id="subset-the-dataset">Subset the dataset</h3>
<p>Because the DESeq2 models will be run on each gradient separately, we need to subset the phyloseq object. This is easily done using <code>HTSSIP::phyloseq_subset</code> <span class="citation" data-cites="youngblut_htssip_2018">(<a href="#ref-youngblut_htssip_2018" role="doc-biblioref">Youngblut, Barnett and Buckley 2018</a>)</span></p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/subset dataset_7f0e9d0a6ea118dd895e7e40e9b1f845">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># split, ignore time points (for labelled ASV plots)</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="do">## IT MIGHT BE POSSIBLE TO OMIT THE LABEL..13C. HERE</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>test_expr_1 <span class="ot">&lt;-</span> <span class="st">"(Species == '${Species}' &amp; Label..13C. == 'Unlabelled') | (Species == '${Species}'  &amp; Label..13C. == '${Label..13C.}')"</span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>params_1 <span class="ot">&lt;-</span> <span class="fu">get_treatment_params</span>(Ps_obj_abs, <span class="fu">c</span>(<span class="st">"Species"</span>,</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Label..13C."</span>),</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Label..13C. != 'Unlabelled'"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `filter_()` was deprecated in dplyr 0.7.0.
ℹ Please use `filter()` instead.
ℹ See vignette('programming') for more help
ℹ The deprecated feature was likely used in the HTSSIP package.
  Please report the issue to the authors.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ps_obj_abs_noTime_l &lt;- phyloseq_subset(subset_samples(Ps_obj_abs, Replicate == 1), params_1, test_expr_1) </span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a list of subsetted phyloseq objects</span></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&gt;%</span> </span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset_samples(Density.zone == "Heavy") %&gt;%</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">phyloseq_subset</span>(., params_1, test_expr_1) <span class="sc">%&gt;%</span> </span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">filter_taxa</span>(.x, <span class="cf">function</span>(y) <span class="fu">sum</span>(y) <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="co"># remove 0-summed ASVs</span></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(., <span class="dv">3</span>) <span class="ot">-&gt;</span> <span class="co"># replicate the list by the number of reps, we will then remove the replicates within each item</span></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_noTime_l</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `mutate_()` was deprecated in dplyr 0.7.0.
ℹ Please use `mutate()` instead.
ℹ See vignette('programming') for more help
ℹ The deprecated feature was likely used in the HTSSIP package.
  Please report the issue to the authors.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only one replicate plus the unlabelled control in each ps obj</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_noTime_l[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Replicate <span class="sc">==</span> <span class="dv">1</span>)) <span class="ot">-&gt;</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_noTime_l_R1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_noTime_l_R1) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"1"</span>)) </span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_noTime_l[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Label..13C. <span class="sc">==</span> <span class="st">"Unlabelled"</span> <span class="sc">|</span> Replicate <span class="sc">==</span> <span class="dv">2</span>)) <span class="ot">-&gt;</span></span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_noTime_l_R2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_noTime_l_R2) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"2"</span>)) </span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_noTime_l[<span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Label..13C. <span class="sc">==</span> <span class="st">"Unlabelled"</span> <span class="sc">|</span> Replicate <span class="sc">==</span> <span class="dv">3</span>)) <span class="ot">-&gt;</span></span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_noTime_l_R3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_noTime_l_R3) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"3"</span>)) </span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a><span class="co"># rejoin all lists</span></span>
<span id="cb168-5"><a href="#cb168-5" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_noTime_l <span class="ot">&lt;-</span> <span class="fu">c</span>(Ps_obj_abs_noTime_l_R1, Ps_obj_abs_noTime_l_R2, Ps_obj_abs_noTime_l_R3)</span>
<span id="cb168-6"><a href="#cb168-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-7"><a href="#cb168-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_noTime_l) <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb168-8"><a href="#cb168-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_remove_all</span>(.x, <span class="st">".*</span><span class="sc">\\</span><span class="st">s</span><span class="sc">\\</span><span class="st">|</span><span class="sc">\\</span><span class="st">s"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb168-9"><a href="#cb168-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_remove_all</span>(.x, <span class="st">"</span><span class="sc">\\</span><span class="st">(|</span><span class="sc">\\</span><span class="st">)|Species == |Label..13C. == |'"</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb168-10"><a href="#cb168-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_replace_all</span>(.x, <span class="st">"</span><span class="sc">\\</span><span class="st">s</span><span class="sc">\\</span><span class="st">s"</span>, <span class="st">" "</span>))</span>
<span id="cb168-11"><a href="#cb168-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># map(., ~str_replace_all(.x, "&amp; ([0-9]+) ", "&amp; \\1 d &amp; ")) </span></span>
<span id="cb168-12"><a href="#cb168-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-13"><a href="#cb168-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Order by name</span></span>
<span id="cb168-14"><a href="#cb168-14" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_noTime_l <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb168-15"><a href="#cb168-15" aria-hidden="true" tabindex="-1"></a>  .[<span class="fu">order</span>(<span class="fu">names</span>(.))]</span>
<span id="cb168-16"><a href="#cb168-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-17"><a href="#cb168-17" aria-hidden="true" tabindex="-1"></a><span class="co"># split, include time points (for DESeq2 modelling) (between gradient comparison)</span></span>
<span id="cb168-18"><a href="#cb168-18" aria-hidden="true" tabindex="-1"></a>test_expr_2 <span class="ot">&lt;-</span> <span class="st">"(Species == '${Species}' &amp;  Day == '${Day}') | (Species == '${Species}' &amp; Day == '${Day}')"</span></span>
<span id="cb168-19"><a href="#cb168-19" aria-hidden="true" tabindex="-1"></a><span class="co"># test_expr_2 &lt;- "(Species == '${Species}' &amp;  Day == '${Day}' &amp; Replicate == '${Replicate}' ) | (Species == '${Species}' &amp; Day == '${Day}' &amp; Replicate == '${Replicate}' )"</span></span>
<span id="cb168-20"><a href="#cb168-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-21"><a href="#cb168-21" aria-hidden="true" tabindex="-1"></a>params_2 <span class="ot">&lt;-</span> <span class="fu">get_treatment_params</span>(Ps_obj_abs, <span class="fu">c</span>(<span class="st">"Species"</span>, </span>
<span id="cb168-22"><a href="#cb168-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"Day"</span>))</span>
<span id="cb168-23"><a href="#cb168-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-24"><a href="#cb168-24" aria-hidden="true" tabindex="-1"></a><span class="co"># params_2 &lt;- get_treatment_params(Ps_obj_abs, c("Species", </span></span>
<span id="cb168-25"><a href="#cb168-25" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    "Day",</span></span>
<span id="cb168-26"><a href="#cb168-26" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    "Replicate"))</span></span>
<span id="cb168-27"><a href="#cb168-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-28"><a href="#cb168-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a list of subsetted phyloseq objects</span></span>
<span id="cb168-29"><a href="#cb168-29" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&gt;%</span> </span>
<span id="cb168-30"><a href="#cb168-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset_samples(Density.zone == "Heavy") %&gt;% </span></span>
<span id="cb168-31"><a href="#cb168-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">phyloseq_subset</span>(., params_2, test_expr_2) <span class="sc">%&gt;%</span> </span>
<span id="cb168-32"><a href="#cb168-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">filter_taxa</span>(.x, <span class="cf">function</span>(y) <span class="fu">sum</span>(y) <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="co"># remove 0-summed ASVs</span></span>
<span id="cb168-33"><a href="#cb168-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(., <span class="dv">3</span>) <span class="ot">-&gt;</span> <span class="co"># replicate the list by the number of reps, we will then remove the replicates within each item</span></span>
<span id="cb168-34"><a href="#cb168-34" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_byTime_l</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only one replicate plus the unlabelled control in each ps obj</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l_R1 <span class="ot">&lt;-</span> Ps_obj_abs_byTime_l[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] <span class="co"># </span></span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l_R1 <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Replicate <span class="sc">==</span> <span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_byTime_l_R1) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"1"</span>)) </span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l_R2 <span class="ot">&lt;-</span> Ps_obj_abs_byTime_l[<span class="dv">9</span><span class="sc">:</span><span class="dv">16</span>]</span>
<span id="cb248-5"><a href="#cb248-5" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l_R2 <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb248-6"><a href="#cb248-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Label..13C. <span class="sc">==</span> <span class="st">"Unlabelled"</span> <span class="sc">|</span> Replicate <span class="sc">==</span> <span class="dv">2</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb312"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb312-1"><a href="#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_byTime_l_R2) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb312-2"><a href="#cb312-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"2"</span>)) </span>
<span id="cb312-3"><a href="#cb312-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-4"><a href="#cb312-4" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l_R3 <span class="ot">&lt;-</span> Ps_obj_abs_byTime_l[<span class="dv">17</span><span class="sc">:</span><span class="dv">24</span>] <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb312-5"><a href="#cb312-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Label..13C. <span class="sc">==</span> <span class="st">"Unlabelled"</span> <span class="sc">|</span> Replicate <span class="sc">==</span> <span class="dv">3</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb376"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb376-1"><a href="#cb376-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_byTime_l_R3) <span class="sc">%&lt;&gt;%</span></span>
<span id="cb376-2"><a href="#cb376-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">paste</span>(.x, <span class="st">"3"</span>)) </span>
<span id="cb376-3"><a href="#cb376-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-4"><a href="#cb376-4" aria-hidden="true" tabindex="-1"></a><span class="co"># rejoin all lists</span></span>
<span id="cb376-5"><a href="#cb376-5" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l <span class="ot">&lt;-</span> <span class="fu">c</span>(Ps_obj_abs_byTime_l_R1, Ps_obj_abs_byTime_l_R2, Ps_obj_abs_byTime_l_R3)</span>
<span id="cb376-6"><a href="#cb376-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-7"><a href="#cb376-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ps_obj_abs_byTime_l) <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb376-8"><a href="#cb376-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_remove_all</span>(.x, <span class="st">".*</span><span class="sc">\\</span><span class="st">s</span><span class="sc">\\</span><span class="st">|</span><span class="sc">\\</span><span class="st">s"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb376-9"><a href="#cb376-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_remove_all</span>(.x, <span class="st">"</span><span class="sc">\\</span><span class="st">(|</span><span class="sc">\\</span><span class="st">)|Species == |Day == |'"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb376-10"><a href="#cb376-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">str_replace_all</span>(.x, <span class="st">"&amp; ([0-9]+) "</span>, <span class="st">"&amp; </span><span class="sc">\\</span><span class="st">1 d &amp; "</span>)) </span>
<span id="cb376-11"><a href="#cb376-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-12"><a href="#cb376-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Order by name</span></span>
<span id="cb376-13"><a href="#cb376-13" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb376-14"><a href="#cb376-14" aria-hidden="true" tabindex="-1"></a>  .[<span class="fu">order</span>(<span class="fu">names</span>(.))]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-abundance-models" class="level3">
<h3 class="anchored" data-anchor-id="differential-abundance-models">Differential abundance models</h3>
<p>Now run the differential abundance models using DESeq2. We then filter the results to include only ASVs with Log_2_ fold change &gt;<code>LFC_thresh</code> and significant at P&lt;<code>alpha_thresh</code>. Lastly, we run ‘LFC-shrinking’ based on Stephens <span class="citation" data-cites="stephens_fdr_2016">(<a href="#ref-stephens_fdr_2016" role="doc-biblioref"><strong>stephens_fdr_2016?</strong></a>)</span>.</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/DESeq2 models by time_3bd68087a8c49dbcc23f745ec01f2bb5">
<div class="sourceCode cell-code" id="cb377"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb377-1"><a href="#cb377-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate a deseq object (see https://bioc.ism.ac.jp/packages/3.3/bioc/vignettes/DESeq2/inst/doc/DESeq2.pdf)</span></span>
<span id="cb377-2"><a href="#cb377-2" aria-hidden="true" tabindex="-1"></a><span class="co"># filter by prevalence</span></span>
<span id="cb377-3"><a href="#cb377-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-4"><a href="#cb377-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Ps_obj_abs_byTime_l &lt;- Ps_obj_abs_byTime_l[1:2]</span></span>
<span id="cb377-5"><a href="#cb377-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-6"><a href="#cb377-6" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l <span class="sc">%&lt;&gt;%</span> <span class="fu">mclapply</span>(.,</span>
<span id="cb377-7"><a href="#cb377-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="cf">function</span>(x) {<span class="fu">filter_ASVs_by_prevalence</span>(<span class="at">ps_obj =</span> x,</span>
<span id="cb377-8"><a href="#cb377-8" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">abund_thresh =</span> abund_thresh,</span>
<span id="cb377-9"><a href="#cb377-9" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">prev_thresh =</span> prev_thresh,</span>
<span id="cb377-10"><a href="#cb377-10" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">var2split =</span> vars2test)},</span>
<span id="cb377-11"><a href="#cb377-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mc.cores =</span> <span class="fu">length</span>(Ps_obj_abs_byTime_l))</span>
<span id="cb377-12"><a href="#cb377-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-13"><a href="#cb377-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only "heavy" fractions</span></span>
<span id="cb377-14"><a href="#cb377-14" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb377-15"><a href="#cb377-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset_samples</span>(.x, Density.zone <span class="sc">==</span> <span class="st">"Heavy"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb377-16"><a href="#cb377-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># map(., ~subset_samples(., Fraction.no. != 2 | Label..13C. != "Labelled")) %&gt;% # remove fraction 2</span></span>
<span id="cb377-17"><a href="#cb377-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">filter_taxa</span>(.x, <span class="cf">function</span>(y) <span class="fu">sum</span>(y) <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="cn">TRUE</span>)) <span class="co"># remove 0-count ASVs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb618"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb618-1"><a href="#cb618-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_byTime_l <span class="sc">%&gt;%</span> </span>
<span id="cb618-2"><a href="#cb618-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mclapply</span>(., <span class="cf">function</span>(x){<span class="fu">ntaxa</span>(x)}) <span class="sc">%&gt;%</span> </span>
<span id="cb618-3"><a href="#cb618-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb618-4"><a href="#cb618-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">everything</span>(), <span class="at">values_to =</span> <span class="st">"ntaxa"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb618-5"><a href="#cb618-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">"name"</span>, <span class="fu">c</span>(Var3, Var2, Var1), <span class="at">sep =</span> <span class="st">" &amp; "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb618-6"><a href="#cb618-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Species) <span class="sc">%&gt;%</span> </span>
<span id="cb618-7"><a href="#cb618-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(ntaxa), <span class="at">se =</span> <span class="fu">sd</span>(ntaxa)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">n</span>())) <span class="sc">%&gt;%</span></span>
<span id="cb618-8"><a href="#cb618-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb618-9"><a href="#cb618-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>, <span class="st">"condensed"</span>, <span class="st">"responsive"</span>), <span class="at">full_width =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Species </th>
   <th style="text-align:right;"> mean </th>
   <th style="text-align:right;"> se </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Epibolus pulchripes </td>
   <td style="text-align:right;"> 889.0833 </td>
   <td style="text-align:right;"> 63.69178 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Glomeris connexa </td>
   <td style="text-align:right;"> 377.3333 </td>
   <td style="text-align:right;"> 24.50520 </td>
  </tr>
</tbody>
</table>

</div>
<div class="sourceCode cell-code" id="cb619"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb619-1"><a href="#cb619-1" aria-hidden="true" tabindex="-1"></a>DESeq_obj_SIP_byTime_l <span class="ot">&lt;-</span> <span class="fu">mclapply</span>(Ps_obj_abs_byTime_l, </span>
<span id="cb619-2"><a href="#cb619-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="cf">function</span>(x) {<span class="fu">phyloseq_to_deseq2_safe</span>(x, </span>
<span id="cb619-3"><a href="#cb619-3" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">test_condition =</span> <span class="st">"Label..13C."</span>, </span>
<span id="cb619-4"><a href="#cb619-4" aria-hidden="true" tabindex="-1"></a>                                                                        <span class="at">ref_level =</span> <span class="st">"Unlabelled"</span>)}, </span>
<span id="cb619-5"><a href="#cb619-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mc.cores =</span> <span class="fu">length</span>(Ps_obj_abs_byTime_l))</span>
<span id="cb619-6"><a href="#cb619-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-7"><a href="#cb619-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-8"><a href="#cb619-8" aria-hidden="true" tabindex="-1"></a><span class="co"># filter_ASVs_by_prevalence &lt;- function (dds = dds_obj, abund = 10, prev = 5) {</span></span>
<span id="cb619-9"><a href="#cb619-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   # Filter ASVs by count number and presence in a DESeq object</span></span>
<span id="cb619-10"><a href="#cb619-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   keep &lt;- rowSums(counts(dds) &gt;= abund) &gt;= prev</span></span>
<span id="cb619-11"><a href="#cb619-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   print(keep)</span></span>
<span id="cb619-12"><a href="#cb619-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   dds &lt;- dds[keep,]</span></span>
<span id="cb619-13"><a href="#cb619-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   return(dds)</span></span>
<span id="cb619-14"><a href="#cb619-14" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb619-15"><a href="#cb619-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb619-16"><a href="#cb619-16" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq_obj_SIP_byTime_l %&lt;&gt;% mclapply(., </span></span>
<span id="cb619-17"><a href="#cb619-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    function(x) {filter_ASVs_by_prevalence(dds = x, </span></span>
<span id="cb619-18"><a href="#cb619-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                                         abund = 10, </span></span>
<span id="cb619-19"><a href="#cb619-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                                         prev = 5)}, </span></span>
<span id="cb619-20"><a href="#cb619-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    mc.cores = length(Ps_obj_abs_byTime_l)) </span></span>
<span id="cb619-21"><a href="#cb619-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-22"><a href="#cb619-22" aria-hidden="true" tabindex="-1"></a><span class="co"># run dds pipeline</span></span>
<span id="cb619-23"><a href="#cb619-23" aria-hidden="true" tabindex="-1"></a>DESeq_obj_SIP_byTime_l <span class="sc">%&lt;&gt;%</span> <span class="fu">mclapply</span>(., </span>
<span id="cb619-24"><a href="#cb619-24" aria-hidden="true" tabindex="-1"></a>                                     <span class="cf">function</span>(x) {<span class="fu">DESeq</span>(x, </span>
<span id="cb619-25"><a href="#cb619-25" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">test =</span> <span class="st">"Wald"</span>, </span>
<span id="cb619-26"><a href="#cb619-26" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">fitType =</span> <span class="st">"parametric"</span>)}, </span>
<span id="cb619-27"><a href="#cb619-27" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mc.cores =</span> <span class="fu">length</span>(Ps_obj_abs_byTime_l)) <span class="co"># run dds pipeline</span></span>
<span id="cb619-28"><a href="#cb619-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-29"><a href="#cb619-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-30"><a href="#cb619-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-31"><a href="#cb619-31" aria-hidden="true" tabindex="-1"></a><span class="co"># extract results from a DESeq analysis</span></span>
<span id="cb619-32"><a href="#cb619-32" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_l  <span class="ot">&lt;-</span> <span class="fu">mclapply</span>(DESeq_obj_SIP_byTime_l, </span>
<span id="cb619-33"><a href="#cb619-33" aria-hidden="true" tabindex="-1"></a>                             <span class="cf">function</span>(x) {</span>
<span id="cb619-34"><a href="#cb619-34" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">results</span>(x, </span>
<span id="cb619-35"><a href="#cb619-35" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">altHypothesis =</span> <span class="st">"greater"</span>,</span>
<span id="cb619-36"><a href="#cb619-36" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">alpha =</span> alpha_thresh, </span>
<span id="cb619-37"><a href="#cb619-37" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"Label..13C."</span>, <span class="st">"Labelled"</span>, <span class="st">"Unlabelled"</span>))}, <span class="co"># redundant if phyloseq_to_deseq2_safe() was used but doesn't hurt</span></span>
<span id="cb619-38"><a href="#cb619-38" aria-hidden="true" tabindex="-1"></a>                             <span class="at">mc.cores =</span> <span class="fu">length</span>(Ps_obj_abs_byTime_l)) </span>
<span id="cb619-39"><a href="#cb619-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-40"><a href="#cb619-40" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_l <span class="ot">&lt;-</span> <span class="fu">mclapply</span>(DESeq_obj_SIP_byTime_l, </span>
<span id="cb619-41"><a href="#cb619-41" aria-hidden="true" tabindex="-1"></a>                                     <span class="cf">function</span>(x) {</span>
<span id="cb619-42"><a href="#cb619-42" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">results</span>(x,</span>
<span id="cb619-43"><a href="#cb619-43" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">lfcThreshold =</span> LFC_thresh,</span>
<span id="cb619-44"><a href="#cb619-44" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">altHypothesis =</span> <span class="st">"greater"</span>,</span>
<span id="cb619-45"><a href="#cb619-45" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">alpha =</span> alpha_thresh,</span>
<span id="cb619-46"><a href="#cb619-46" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"Label..13C."</span>, <span class="st">"Labelled"</span>, <span class="st">"Unlabelled"</span>))}, <span class="co"># redundant if phyloseq_to_deseq2_safe() was used but doesn't hurt</span></span>
<span id="cb619-47"><a href="#cb619-47" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mc.cores =</span> <span class="fu">length</span>(Ps_obj_abs_byTime_l)) <span class="co"># Extract results from a DESeq analysis</span></span>
<span id="cb619-48"><a href="#cb619-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-49"><a href="#cb619-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb619-50"><a href="#cb619-50" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_shrink_l <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">length</span>(DESeq_obj_SIP_byTime_l)), </span>
<span id="cb619-51"><a href="#cb619-51" aria-hidden="true" tabindex="-1"></a>                                             <span class="sc">~</span><span class="fu">lfcShrink</span>(DESeq_obj_SIP_byTime_l[[.x]],</span>
<span id="cb619-52"><a href="#cb619-52" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">res =</span> DESeq_res_SIP_byTime_LFC_l[[.x]],</span>
<span id="cb619-53"><a href="#cb619-53" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">coef =</span> <span class="st">"Label..13C._Labelled_vs_Unlabelled"</span>,</span>
<span id="cb619-54"><a href="#cb619-54" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">type =</span> <span class="st">"apeglm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in nbinomGLM(x = x, Y = YNZ, size = size, weights = weightsNZ, offset =
offsetNZ, : the moving direction increases the objective function value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>some rows did not converge in finding the MAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb626"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb626-1"><a href="#cb626-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(DESeq_res_SIP_byTime_LFC_shrink_l) <span class="ot">&lt;-</span> <span class="fu">names</span>(DESeq_res_SIP_byTime_LFC_l)</span>
<span id="cb626-2"><a href="#cb626-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb626-3"><a href="#cb626-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare</span></span>
<span id="cb626-4"><a href="#cb626-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(DESeq_res_SIP_byTime_l[[<span class="dv">2</span>]], <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>))</span>
<span id="cb626-5"><a href="#cb626-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(DESeq_res_SIP_byTime_LFC_l[[<span class="dv">2</span>]], <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/DESeq2%20models%20by%20time-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb627"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb627-1"><a href="#cb627-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(DESeq_res_SIP_byTime_LFC_shrink_l[[<span class="dv">2</span>]], <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/DESeq2%20models%20by%20time-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb628"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb628-1"><a href="#cb628-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summarise results (lfcShrink doesn't change the values)</span></span>
<span id="cb628-2"><a href="#cb628-2" aria-hidden="true" tabindex="-1"></a><span class="co"># map2(DESeq_res_SIP_byTime_l, print(names(DESeq_res_SIP_byTime_l)), ~summary(.x)) # summarise results</span></span>
<span id="cb628-3"><a href="#cb628-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(DESeq_res_SIP_byTime_l))) { <span class="co"># didn't manage with map</span></span>
<span id="cb628-4"><a href="#cb628-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">names</span>(DESeq_res_SIP_byTime_l[i]))</span>
<span id="cb628-5"><a href="#cb628-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(DESeq_res_SIP_byTime_l[[i]])</span>
<span id="cb628-6"><a href="#cb628-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Epibolus pulchripes &amp; 14 d &amp; 1"

out of 1048 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 193, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 102, 9.7%
low counts [2]     : 158, 15%
(mean count &lt; 25)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 14 d &amp; 2"

out of 1067 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 203, 19%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 107, 10%
low counts [2]     : 140, 13%
(mean count &lt; 23)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 14 d &amp; 3"

out of 1066 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 84, 7.9%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 97, 9.1%
low counts [2]     : 108, 10%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 1"

out of 1221 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 225, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 94, 7.7%
low counts [2]     : 117, 9.6%
(mean count &lt; 57)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 2"

out of 843 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 128, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 99, 12%
low counts [2]     : 0, 0%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 3"

out of 383 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 68, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 36, 9.4%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 1"

out of 755 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 201, 27%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 14, 1.9%
low counts [2]     : 164, 22%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 2"

out of 908 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 178, 20%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 37, 4.1%
low counts [2]     : 237, 26%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 3"

out of 819 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 70, 8.5%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 141, 17%
low counts [2]     : 298, 36%
(mean count &lt; 16)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 1"

out of 864 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 102, 12%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 92, 11%
low counts [2]     : 286, 33%
(mean count &lt; 11)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 2"

out of 687 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 182, 26%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 54, 7.9%
low counts [2]     : 14, 2%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 3"

out of 1008 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 139, 14%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 131, 13%
low counts [2]     : 225, 22%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 1"

out of 264 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 38, 14%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 37, 14%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 2"

out of 369 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 116, 31%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 25, 6.8%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 3"

out of 366 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 97, 27%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 20, 5.5%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 1"

out of 454 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 96, 21%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 48, 11%
low counts [2]     : 47, 10%
(mean count &lt; 101)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 2"

out of 469 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 108, 23%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 39, 8.3%
low counts [2]     : 0, 0%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 3"

out of 238 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 20, 8.4%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 36, 15%
low counts [2]     : 148, 62%
(mean count &lt; 1583)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 1"

out of 363 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 63, 17%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 29, 8%
low counts [2]     : 28, 7.7%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 2"

out of 508 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 64, 13%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 82, 16%
low counts [2]     : 40, 7.9%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 3"

out of 450 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 68, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 34, 7.6%
low counts [2]     : 68, 15%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 1"

out of 300 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 45, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 26, 8.7%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 2"

out of 329 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 58, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 33, 10%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 3"

out of 418 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 25, 6%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 100, 24%
low counts [2]     : 0, 0%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb630"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb630-1"><a href="#cb630-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(DESeq_res_SIP_byTime_LFC_l))) { <span class="co"># didn't manage with map</span></span>
<span id="cb630-2"><a href="#cb630-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">names</span>(DESeq_res_SIP_byTime_LFC_l[i]))</span>
<span id="cb630-3"><a href="#cb630-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(DESeq_res_SIP_byTime_LFC_l[[i]])</span>
<span id="cb630-4"><a href="#cb630-4" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Epibolus pulchripes &amp; 14 d &amp; 1"

out of 1048 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 193, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 102, 9.7%
low counts [2]     : 158, 15%
(mean count &lt; 25)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 14 d &amp; 2"

out of 1067 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 203, 19%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 107, 10%
low counts [2]     : 140, 13%
(mean count &lt; 23)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 14 d &amp; 3"

out of 1066 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 84, 7.9%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 97, 9.1%
low counts [2]     : 108, 10%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 1"

out of 1221 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 225, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 94, 7.7%
low counts [2]     : 117, 9.6%
(mean count &lt; 57)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 2"

out of 843 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 128, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 99, 12%
low counts [2]     : 0, 0%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 21 d &amp; 3"

out of 383 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 68, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 36, 9.4%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 1"

out of 755 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 201, 27%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 14, 1.9%
low counts [2]     : 164, 22%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 2"

out of 908 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 178, 20%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 37, 4.1%
low counts [2]     : 237, 26%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 3 d &amp; 3"

out of 819 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 70, 8.5%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 141, 17%
low counts [2]     : 298, 36%
(mean count &lt; 16)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 1"

out of 864 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 102, 12%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 92, 11%
low counts [2]     : 286, 33%
(mean count &lt; 11)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 2"

out of 687 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 182, 26%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 54, 7.9%
low counts [2]     : 14, 2%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Epibolus pulchripes &amp; 7 d &amp; 3"

out of 1008 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 139, 14%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 131, 13%
low counts [2]     : 225, 22%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 1"

out of 264 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 38, 14%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 37, 14%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 2"

out of 369 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 116, 31%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 25, 6.8%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 14 d &amp; 3"

out of 366 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 97, 27%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 20, 5.5%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 1"

out of 454 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 96, 21%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 48, 11%
low counts [2]     : 47, 10%
(mean count &lt; 101)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 2"

out of 469 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 108, 23%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 39, 8.3%
low counts [2]     : 0, 0%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 21 d &amp; 3"

out of 238 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 20, 8.4%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 36, 15%
low counts [2]     : 148, 62%
(mean count &lt; 1583)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 1"

out of 363 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 63, 17%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 29, 8%
low counts [2]     : 28, 7.7%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 2"

out of 508 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 64, 13%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 82, 16%
low counts [2]     : 40, 7.9%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 3 d &amp; 3"

out of 450 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 68, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 34, 7.6%
low counts [2]     : 68, 15%
(mean count &lt; 5)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 1"

out of 300 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 45, 15%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 26, 8.7%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 2"

out of 329 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 58, 18%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 33, 10%
low counts [2]     : 0, 0%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

[1] "Glomeris connexa &amp; 7 d &amp; 3"

out of 418 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 25, 6%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 100, 24%
low counts [2]     : 0, 0%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb632"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb632-1"><a href="#cb632-1" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_shrink_l <span class="sc">%&lt;&gt;%</span> <span class="fu">mclapply</span>(., <span class="cf">function</span>(x) {<span class="fu">mark_sig_DESeq_ASVs</span>(x,</span>
<span id="cb632-2"><a href="#cb632-2" aria-hidden="true" tabindex="-1"></a>                                                                 <span class="at">two_way =</span> F,</span>
<span id="cb632-3"><a href="#cb632-3" aria-hidden="true" tabindex="-1"></a>                                                                 <span class="at">sig_level =</span> alpha_thresh, </span>
<span id="cb632-4"><a href="#cb632-4" aria-hidden="true" tabindex="-1"></a>                                                                 <span class="at">LFC =</span> LFC_thresh, </span>
<span id="cb632-5"><a href="#cb632-5" aria-hidden="true" tabindex="-1"></a>                                                                 <span class="at">vars2test =</span> <span class="st">"Label..13C."</span>)})</span>
<span id="cb632-6"><a href="#cb632-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb632-7"><a href="#cb632-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Store labelled ASVs and save them to a file</span></span>
<span id="cb632-8"><a href="#cb632-8" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_shrink_l <span class="sc">%&gt;%</span> </span>
<span id="cb632-9"><a href="#cb632-9" aria-hidden="true" tabindex="-1"></a>   <span class="co"># map(., ~subset(.x, padj &lt; alpha_thresh &amp; log2FoldChange &gt; LFC_thresh)) %&gt;% </span></span>
<span id="cb632-10"><a href="#cb632-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">subset</span>(.x, Significance <span class="sc">==</span> <span class="st">"Pass"</span>)) <span class="sc">%&gt;%</span> <span class="co"># only significant ones</span></span>
<span id="cb632-11"><a href="#cb632-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">as.data.frame</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-12"><a href="#cb632-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="st">"ASV"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-13"><a href="#cb632-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(., <span class="at">.id =</span> <span class="st">"Comparison"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb632-14"><a href="#cb632-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Comparison, <span class="fu">desc</span>(baseMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-15"><a href="#cb632-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># arrange(Comparison, desc(baseMean)) %&gt;% </span></span>
<span id="cb632-16"><a href="#cb632-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(., <span class="st">"Comparison"</span> ,<span class="fu">c</span>(Var3, Var2, Var1), <span class="at">sep =</span> <span class="st">" &amp; "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb632-17"><a href="#cb632-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter the results and keep only ASVs that appear in at least prev_thresh replicate from each species </span></span>
<span id="cb632-18"><a href="#cb632-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ASV, <span class="sc">!!</span><span class="fu">sym</span>(Var3)) <span class="sc">%&gt;%</span> <span class="co"># group by species</span></span>
<span id="cb632-19"><a href="#cb632-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ASV_count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="co"># count ASV occurance </span></span>
<span id="cb632-20"><a href="#cb632-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV_count <span class="sc">&gt;</span> prev_thresh) <span class="sc">%&gt;%</span> <span class="co"># keep only those that appear in more than prev_thresh samples per species</span></span>
<span id="cb632-21"><a href="#cb632-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%T&gt;%</span> </span>
<span id="cb632-22"><a href="#cb632-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(., <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"DESeq_byTime_a-"</span>, alpha_thresh, <span class="st">"_"</span>, LFC_thresh, <span class="st">"_prev.tsv"</span>)) <span class="ot">-&gt;</span></span>
<span id="cb632-23"><a href="#cb632-23" aria-hidden="true" tabindex="-1"></a>  DESeq_res_SIP_byTime_sig_prev_df</span>
<span id="cb632-24"><a href="#cb632-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb632-25"><a href="#cb632-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a new list of DESeq2 results with means across replicates</span></span>
<span id="cb632-26"><a href="#cb632-26" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_shrink_l <span class="sc">%&gt;%</span> </span>
<span id="cb632-27"><a href="#cb632-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># map(., ~subset(.x, padj &lt; alpha_thresh &amp; log2FoldChange &gt; LFC_thresh)) %&gt;% </span></span>
<span id="cb632-28"><a href="#cb632-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">as.data.frame</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-29"><a href="#cb632-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="st">"ASV"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-30"><a href="#cb632-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(., <span class="at">.id =</span> <span class="st">"Comparison"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb632-31"><a href="#cb632-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Comparison, <span class="fu">desc</span>(baseMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb632-32"><a href="#cb632-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># arrange(Comparison, desc(baseMean)) %&gt;% </span></span>
<span id="cb632-33"><a href="#cb632-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(., <span class="st">"Comparison"</span> ,<span class="fu">c</span>(Var3, Var2, Var1), <span class="at">sep =</span> <span class="st">" &amp; "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb632-34"><a href="#cb632-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ASV, <span class="sc">!!</span><span class="fu">sym</span>(Var3)) <span class="sc">%&gt;%</span> <span class="co"># group by species</span></span>
<span id="cb632-35"><a href="#cb632-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ASV_count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="co"># count ASV occurrence </span></span>
<span id="cb632-36"><a href="#cb632-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sig_log =</span> <span class="fu">if_else</span>(Significance <span class="sc">==</span> <span class="st">"Pass"</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span>  <span class="co"># make a logical variable from significance</span></span>
<span id="cb632-37"><a href="#cb632-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Species, Day, ASV) <span class="sc">%&gt;%</span></span>
<span id="cb632-38"><a href="#cb632-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">log2FoldChange =</span> <span class="fu">mean</span>(log2FoldChange), </span>
<span id="cb632-39"><a href="#cb632-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">lfcSE =</span> <span class="fu">mean</span>(lfcSE, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb632-40"><a href="#cb632-40" aria-hidden="true" tabindex="-1"></a>            <span class="at">pvalue =</span> <span class="fu">min</span>(pvalue, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb632-41"><a href="#cb632-41" aria-hidden="true" tabindex="-1"></a>            <span class="at">padj =</span> <span class="fu">min</span>(padj, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb632-42"><a href="#cb632-42" aria-hidden="true" tabindex="-1"></a>            <span class="at">ASV_count =</span> <span class="fu">mean</span>(ASV_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb632-43"><a href="#cb632-43" aria-hidden="true" tabindex="-1"></a>            <span class="at">count_per_group =</span> <span class="fu">n</span>(), <span class="co"># this is ASV count but per group from group_by()</span></span>
<span id="cb632-44"><a href="#cb632-44" aria-hidden="true" tabindex="-1"></a>            <span class="at">Significance =</span> <span class="fu">if_else</span>(purrr<span class="sc">::</span><span class="fu">reduce</span>(sig_log, <span class="st">`</span><span class="at">+</span><span class="st">`</span>) <span class="sc">&gt;=</span> rep_thresh, <span class="st">"Pass"</span>, <span class="st">"Fail"</span>)</span>
<span id="cb632-45"><a href="#cb632-45" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> <span class="co"># only keep ASV that are significant in &gt;= rep_thresh replicates per group</span></span>
<span id="cb632-46"><a href="#cb632-46" aria-hidden="true" tabindex="-1"></a>             <span class="co"># the names need to be fixed with glue</span></span>
<span id="cb632-47"><a href="#cb632-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Significance = if_else(get(paste(vars2test, "- P adj.")) &lt; sig_level &amp;</span></span>
<span id="cb632-48"><a href="#cb632-48" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                                      !is.na(get(paste(vars2test, "- P adj."))) &amp;</span></span>
<span id="cb632-49"><a href="#cb632-49" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                                      abs(`log2 fold change`) &gt; LFC &amp;</span></span>
<span id="cb632-50"><a href="#cb632-50" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                                 ASV_count &gt; prev_thresh, "Pass", "Fail")) %&gt;% # sig ASV only those that appear in more than prev_thresh gradients per species</span></span>
<span id="cb632-51"><a href="#cb632-51" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb632-52"><a href="#cb632-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Species, Day) <span class="sc">%&gt;%</span> </span>
<span id="cb632-53"><a href="#cb632-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># left_join(., </span></span>
<span id="cb632-54"><a href="#cb632-54" aria-hidden="true" tabindex="-1"></a>            <span class="co"># dplyr::select(distinct(DESeq_res_SIP_byTime_sig_prev_df, Species, Day, ASV, .keep_all= TRUE), Species, Day, ASV, Significance), </span></span>
<span id="cb632-55"><a href="#cb632-55" aria-hidden="true" tabindex="-1"></a>            <span class="co"># join_by(Species, Day, ASV), keep = FALSE) %&gt;% </span></span>
<span id="cb632-56"><a href="#cb632-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Significance = replace_na(Significance, "Fail")) %T&gt;%</span></span>
<span id="cb632-57"><a href="#cb632-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(., <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"DESeq_byTime_a-"</span>, alpha_thresh, <span class="st">"_"</span>, LFC_thresh, <span class="st">"_prev_concensus.tsv"</span>)) <span class="sc">%&gt;%</span> <span class="co"># save to file</span></span>
<span id="cb632-58"><a href="#cb632-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="st">"DESeq_res_SIP_byTime_LFC_sig_prev_con_df"</span>, ., <span class="at">envir =</span> .GlobalEnv) <span class="sc">%&gt;%</span> <span class="co"># also save to data frame</span></span>
<span id="cb632-59"><a href="#cb632-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Species, Day) <span class="sc">%&gt;%</span> </span>
<span id="cb632-60"><a href="#cb632-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_split</span>() <span class="ot">-&gt;</span></span>
<span id="cb632-61"><a href="#cb632-61" aria-hidden="true" tabindex="-1"></a>  DESeq_res_SIP_byTime_LFC_sig_prev_con_l</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 1991 warnings in `summarise()`.
The first warning was:
ℹ In argument: `pvalue = min(pvalue, na.rm = TRUE)`.
ℹ In group 9: `Species = "Epibolus pulchripes"`, `Day = "14 d"`, `ASV =
  "Seq_1016"`.
Caused by warning in `min()`:
! no non-missing arguments to min; returning Inf
ℹ Run `dplyr::last_dplyr_warnings()` to see the 1990 remaining warnings.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Species', 'Day'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb635"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb635-1"><a href="#cb635-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb635-2"><a href="#cb635-2" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_sig_prev_con_l <span class="sc">%&gt;%</span></span>
<span id="cb635-3"><a href="#cb635-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">transmute</span>(.x, <span class="at">name =</span> <span class="fu">paste</span>(Species, Day))[<span class="dv">1</span>, ]) <span class="sc">%&gt;%</span> </span>
<span id="cb635-4"><a href="#cb635-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">pull</span>(.x)) <span class="ot">-&gt;</span> </span>
<span id="cb635-5"><a href="#cb635-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># map(., ~str_remove(.x, " &amp; [0-9]$")) -&gt;</span></span>
<span id="cb635-6"><a href="#cb635-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(DESeq_res_SIP_byTime_LFC_sig_prev_con_l)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="inspect-results" class="level4">
<h4 class="anchored" data-anchor-id="inspect-results">Inspect results</h4>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/vis DES res_77127151a7432b1c480f5034c648e42d">
<div class="sourceCode cell-code" id="cb636"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb636-1"><a href="#cb636-1" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb636-2"><a href="#cb636-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get_variable() %&gt;%</span></span>
<span id="cb636-3"><a href="#cb636-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">%&gt;%</span> </span>
<span id="cb636-4"><a href="#cb636-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">vis_value</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/vis%20DES%20res-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb637"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb637-1"><a href="#cb637-1" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb637-2"><a href="#cb637-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get_variable() %&gt;% </span></span>
<span id="cb637-3"><a href="#cb637-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(is.numeric) <span class="sc">%&gt;%</span> </span>
<span id="cb637-4"><a href="#cb637-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">vis_cor</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/vis%20DES%20res-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="plot-differential-abundance-models" class="level4">
<h4 class="anchored" data-anchor-id="plot-differential-abundance-models">Plot differential abundance models</h4>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/plot DESeq2 models_161dd1c84c25143c8e250baed702f82c">
<div class="sourceCode cell-code" id="cb638"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb638-1"><a href="#cb638-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ps_obj &lt;- Ps_obj_abs</span></span>
<span id="cb638-2"><a href="#cb638-2" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq_results &lt;- DESeq_res_SIP_byTime_LFC_l[12]</span></span>
<span id="cb638-3"><a href="#cb638-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_DESeq(DESeq_results, ps_obj, plot_title = names(DESeq_results))</span></span>
<span id="cb638-4"><a href="#cb638-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb638-5"><a href="#cb638-5" aria-hidden="true" tabindex="-1"></a>DESeq_plots <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">length</span>(DESeq_res_SIP_byTime_LFC_shrink_l)), </span>
<span id="cb638-6"><a href="#cb638-6" aria-hidden="true" tabindex="-1"></a>                        <span class="sc">~</span><span class="fu">plot_DESeq</span>(DESeq_res_SIP_byTime_LFC_shrink_l[.x],</span>
<span id="cb638-7"><a href="#cb638-7" aria-hidden="true" tabindex="-1"></a>                                    Ps_obj_abs, <span class="at">plot_title =</span> <span class="fu">names</span>(DESeq_res_SIP_byTime_LFC_shrink_l[.x])))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `group_by()`.
ℹ In argument: `across(rank)`.
Caused by warning:
! Using an external vector in selections was deprecated in tidyselect 1.1.0.
ℹ Please use `all_of()` or `any_of()` instead.
  # Was:
  data %&gt;% select(rank)

  # Now:
  data %&gt;% select(all_of(rank))

See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1461"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1461-1"><a href="#cb1461-1" aria-hidden="true" tabindex="-1"></a>Epibolus_DESeq <span class="ot">&lt;-</span> ((DESeq_plots[[<span class="dv">7</span>]] <span class="sc">+</span> </span>
<span id="cb1461-2"><a href="#cb1461-2" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb1461-3"><a href="#cb1461-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-4"><a href="#cb1461-4" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-5"><a href="#cb1461-5" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">8</span>]] <span class="sc">+</span> </span>
<span id="cb1461-6"><a href="#cb1461-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-7"><a href="#cb1461-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-8"><a href="#cb1461-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-9"><a href="#cb1461-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-10"><a href="#cb1461-10" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">9</span>]] <span class="sc">+</span> </span>
<span id="cb1461-11"><a href="#cb1461-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-12"><a href="#cb1461-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-13"><a href="#cb1461-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-14"><a href="#cb1461-14" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-15"><a href="#cb1461-15" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">10</span>]] <span class="sc">+</span> </span>
<span id="cb1461-16"><a href="#cb1461-16" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-17"><a href="#cb1461-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-18"><a href="#cb1461-18" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-19"><a href="#cb1461-19" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">11</span>]] <span class="sc">+</span> </span>
<span id="cb1461-20"><a href="#cb1461-20" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-21"><a href="#cb1461-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-22"><a href="#cb1461-22" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-23"><a href="#cb1461-23" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-24"><a href="#cb1461-24" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">12</span>]] <span class="sc">+</span> </span>
<span id="cb1461-25"><a href="#cb1461-25" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-26"><a href="#cb1461-26" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-27"><a href="#cb1461-27" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-28"><a href="#cb1461-28" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-29"><a href="#cb1461-29" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">1</span>]] <span class="sc">+</span> </span>
<span id="cb1461-30"><a href="#cb1461-30" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-31"><a href="#cb1461-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-32"><a href="#cb1461-32" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-33"><a href="#cb1461-33" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">2</span>]] <span class="sc">+</span> </span>
<span id="cb1461-34"><a href="#cb1461-34" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1461-35"><a href="#cb1461-35" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-36"><a href="#cb1461-36" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-37"><a href="#cb1461-37" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-38"><a href="#cb1461-38" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">3</span>]] <span class="sc">+</span> </span>
<span id="cb1461-39"><a href="#cb1461-39" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-40"><a href="#cb1461-40" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1461-41"><a href="#cb1461-41" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-42"><a href="#cb1461-42" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-43"><a href="#cb1461-43" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">4</span>]] <span class="sc">+</span> </span>
<span id="cb1461-44"><a href="#cb1461-44" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb1461-45"><a href="#cb1461-45" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-46"><a href="#cb1461-46" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">5</span>]] <span class="sc">+</span> </span>
<span id="cb1461-47"><a href="#cb1461-47" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1461-48"><a href="#cb1461-48" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-49"><a href="#cb1461-49" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1461-50"><a href="#cb1461-50" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">6</span>]] <span class="sc">+</span> </span>
<span id="cb1461-51"><a href="#cb1461-51" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1461-52"><a href="#cb1461-52" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1461-53"><a href="#cb1461-53" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>))<span class="sc">+</span></span>
<span id="cb1461-54"><a href="#cb1461-54" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">guides =</span> <span class="st">"collect"</span>) <span class="sc">&amp;</span> </span>
<span id="cb1461-55"><a href="#cb1461-55" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'bottom'</span>))</span>
<span id="cb1461-56"><a href="#cb1461-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1461-57"><a href="#cb1461-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1461-58"><a href="#cb1461-58" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Epibolus_all_DESeq2"</span>), </span>
<span id="cb1461-59"><a href="#cb1461-59" aria-hidden="true" tabindex="-1"></a>            Epibolus_DESeq, </span>
<span id="cb1461-60"><a href="#cb1461-60" aria-hidden="true" tabindex="-1"></a>            <span class="at">pwidth =</span> <span class="dv">14</span>, </span>
<span id="cb1461-61"><a href="#cb1461-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">pheight =</span> <span class="dv">12</span>,</span>
<span id="cb1461-62"><a href="#cb1461-62" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 554 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 730 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 749 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 762 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 505 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 869 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 855 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 864 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 982 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 996 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 715 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 315 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 554 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 730 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 749 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 762 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 505 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 869 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 855 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 864 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 982 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 996 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 715 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 315 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1486"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1486-1"><a href="#cb1486-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Epibolus_all_DESeq2"</span>, <span class="st">".png"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures/Epibolus_all_DESeq2.png" class="img-fluid" width="4200"></p>
</div>
<div class="sourceCode cell-code" id="cb1487"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1487-1"><a href="#cb1487-1" aria-hidden="true" tabindex="-1"></a>Glomeris_DESeq <span class="ot">&lt;-</span> ((DESeq_plots[[<span class="dv">19</span>]] <span class="sc">+</span> </span>
<span id="cb1487-2"><a href="#cb1487-2" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb1487-3"><a href="#cb1487-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-4"><a href="#cb1487-4" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-5"><a href="#cb1487-5" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">20</span>]] <span class="sc">+</span> </span>
<span id="cb1487-6"><a href="#cb1487-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-7"><a href="#cb1487-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-8"><a href="#cb1487-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-9"><a href="#cb1487-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-10"><a href="#cb1487-10" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">21</span>]] <span class="sc">+</span> </span>
<span id="cb1487-11"><a href="#cb1487-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-12"><a href="#cb1487-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-13"><a href="#cb1487-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-14"><a href="#cb1487-14" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-15"><a href="#cb1487-15" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">22</span>]] <span class="sc">+</span> </span>
<span id="cb1487-16"><a href="#cb1487-16" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-17"><a href="#cb1487-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-18"><a href="#cb1487-18" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-19"><a href="#cb1487-19" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">23</span>]] <span class="sc">+</span> </span>
<span id="cb1487-20"><a href="#cb1487-20" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-21"><a href="#cb1487-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-22"><a href="#cb1487-22" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-23"><a href="#cb1487-23" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-24"><a href="#cb1487-24" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">24</span>]] <span class="sc">+</span> </span>
<span id="cb1487-25"><a href="#cb1487-25" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-26"><a href="#cb1487-26" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-27"><a href="#cb1487-27" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-28"><a href="#cb1487-28" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-29"><a href="#cb1487-29" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">13</span>]] <span class="sc">+</span> </span>
<span id="cb1487-30"><a href="#cb1487-30" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-31"><a href="#cb1487-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-32"><a href="#cb1487-32" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-33"><a href="#cb1487-33" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">14</span>]] <span class="sc">+</span> </span>
<span id="cb1487-34"><a href="#cb1487-34" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1487-35"><a href="#cb1487-35" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-36"><a href="#cb1487-36" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-37"><a href="#cb1487-37" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-38"><a href="#cb1487-38" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">15</span>]] <span class="sc">+</span> </span>
<span id="cb1487-39"><a href="#cb1487-39" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-40"><a href="#cb1487-40" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1487-41"><a href="#cb1487-41" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-42"><a href="#cb1487-42" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-43"><a href="#cb1487-43" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">16</span>]] <span class="sc">+</span> </span>
<span id="cb1487-44"><a href="#cb1487-44" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb1487-45"><a href="#cb1487-45" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-46"><a href="#cb1487-46" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">17</span>]] <span class="sc">+</span> </span>
<span id="cb1487-47"><a href="#cb1487-47" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1487-48"><a href="#cb1487-48" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-49"><a href="#cb1487-49" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-50"><a href="#cb1487-50" aria-hidden="true" tabindex="-1"></a>                     (DESeq_plots[[<span class="dv">18</span>]] <span class="sc">+</span> </span>
<span id="cb1487-51"><a href="#cb1487-51" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1487-52"><a href="#cb1487-52" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1487-53"><a href="#cb1487-53" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">30</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb1487-54"><a href="#cb1487-54" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">guides =</span> <span class="st">"collect"</span>) <span class="sc">&amp;</span> </span>
<span id="cb1487-55"><a href="#cb1487-55" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'bottom'</span>))</span>
<span id="cb1487-56"><a href="#cb1487-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1487-57"><a href="#cb1487-57" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Glomeris_DESeq2"</span>), </span>
<span id="cb1487-58"><a href="#cb1487-58" aria-hidden="true" tabindex="-1"></a>            Glomeris_DESeq, </span>
<span id="cb1487-59"><a href="#cb1487-59" aria-hidden="true" tabindex="-1"></a>            <span class="at">pwidth =</span> <span class="dv">14</span>, </span>
<span id="cb1487-60"><a href="#cb1487-60" aria-hidden="true" tabindex="-1"></a>            <span class="at">pheight =</span> <span class="dv">12</span>,</span>
<span id="cb1487-61"><a href="#cb1487-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 300 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 444 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 382 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 255 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 272 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 393 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 226 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 253 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 269 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 358 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 361 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 218 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 300 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 444 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 382 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 255 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 272 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 393 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 226 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 253 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 269 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 358 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 361 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 218 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1512"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1512-1"><a href="#cb1512-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Glomeris_DESeq2"</span>, <span class="st">".png"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures/Glomeris_DESeq2.png" class="img-fluid" width="4200"></p>
</div>
<div class="sourceCode cell-code" id="cb1513"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1513-1"><a href="#cb1513-1" aria-hidden="true" tabindex="-1"></a>DESeq_plots_con <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">length</span>(DESeq_res_SIP_byTime_LFC_sig_prev_con_l)), </span>
<span id="cb1513-2"><a href="#cb1513-2" aria-hidden="true" tabindex="-1"></a>                        <span class="sc">~</span><span class="fu">plot_DESeq</span>(DESeq_res_SIP_byTime_LFC_sig_prev_con_l[.x],</span>
<span id="cb1513-3"><a href="#cb1513-3" aria-hidden="true" tabindex="-1"></a>                                    Ps_obj_abs, <span class="at">plot_title =</span> <span class="st">""</span>, <span class="at">plot_empty =</span> <span class="st">"TRUE"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1785"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1785-1"><a href="#cb1785-1" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="st">"</span></span>
<span id="cb1785-2"><a href="#cb1785-2" aria-hidden="true" tabindex="-1"></a><span class="st">1234</span></span>
<span id="cb1785-3"><a href="#cb1785-3" aria-hidden="true" tabindex="-1"></a><span class="st">5678</span></span>
<span id="cb1785-4"><a href="#cb1785-4" aria-hidden="true" tabindex="-1"></a><span class="st">9999</span></span>
<span id="cb1785-5"><a href="#cb1785-5" aria-hidden="true" tabindex="-1"></a><span class="st">"</span></span>
<span id="cb1785-6"><a href="#cb1785-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1785-7"><a href="#cb1785-7" aria-hidden="true" tabindex="-1"></a>All_DESeq <span class="ot">&lt;-</span> ((DESeq_plots_con[[<span class="dv">3</span>]] <span class="sc">+</span> </span>
<span id="cb1785-8"><a href="#cb1785-8" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1785-9"><a href="#cb1785-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-10"><a href="#cb1785-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1785-11"><a href="#cb1785-11" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1785-12"><a href="#cb1785-12" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">facet_grid</span>(. <span class="sc">~</span> Day)) <span class="sc">+</span></span>
<span id="cb1785-13"><a href="#cb1785-13" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">4</span>]] <span class="sc">+</span> </span>
<span id="cb1785-14"><a href="#cb1785-14" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1785-15"><a href="#cb1785-15" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-16"><a href="#cb1785-16" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1785-17"><a href="#cb1785-17" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1785-18"><a href="#cb1785-18" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">facet_grid</span>(. <span class="sc">~</span> Day)) <span class="sc">+</span></span>
<span id="cb1785-19"><a href="#cb1785-19" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">1</span>]] <span class="sc">+</span> </span>
<span id="cb1785-20"><a href="#cb1785-20" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1785-21"><a href="#cb1785-21" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-22"><a href="#cb1785-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1785-23"><a href="#cb1785-23" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1785-24"><a href="#cb1785-24" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">facet_grid</span>(. <span class="sc">~</span> Day)) <span class="sc">+</span></span>
<span id="cb1785-25"><a href="#cb1785-25" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">2</span>]] <span class="sc">+</span> </span>
<span id="cb1785-26"><a href="#cb1785-26" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1785-27"><a href="#cb1785-27" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-28"><a href="#cb1785-28" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-29"><a href="#cb1785-29" aria-hidden="true" tabindex="-1"></a>                           <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb1785-30"><a href="#cb1785-30" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1785-31"><a href="#cb1785-31" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">facet_grid</span>(Species <span class="sc">~</span> Day)) <span class="sc">+</span></span>
<span id="cb1785-32"><a href="#cb1785-32" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">7</span>]] <span class="sc">+</span> </span>
<span id="cb1785-33"><a href="#cb1785-33" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb1785-34"><a href="#cb1785-34" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb1785-35"><a href="#cb1785-35" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">8</span>]] <span class="sc">+</span> </span>
<span id="cb1785-36"><a href="#cb1785-36" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1785-37"><a href="#cb1785-37" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1785-38"><a href="#cb1785-38" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb1785-39"><a href="#cb1785-39" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">5</span>]] <span class="sc">+</span> </span>
<span id="cb1785-40"><a href="#cb1785-40" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1785-41"><a href="#cb1785-41" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb1785-42"><a href="#cb1785-42" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb1785-43"><a href="#cb1785-43" aria-hidden="true" tabindex="-1"></a>                  (DESeq_plots_con[[<span class="dv">6</span>]] <span class="sc">+</span> </span>
<span id="cb1785-44"><a href="#cb1785-44" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb1785-45"><a href="#cb1785-45" aria-hidden="true" tabindex="-1"></a>                           <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1785-46"><a href="#cb1785-46" aria-hidden="true" tabindex="-1"></a>                           <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb1785-47"><a href="#cb1785-47" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1785-48"><a href="#cb1785-48" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">facet_grid</span>(Species <span class="sc">~</span> .)) <span class="sc">+</span></span>
<span id="cb1785-49"><a href="#cb1785-49" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">as_ggplot</span>(<span class="fu">get_legend</span>(DESeq_plots_con[[<span class="dv">1</span>]]))) <span class="sc">+</span></span>
<span id="cb1785-50"><a href="#cb1785-50" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">plot_layout</span>(<span class="at">design =</span> design, <span class="at">guides =</span> <span class="st">"collect"</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1126 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1787"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1787-1"><a href="#cb1787-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_figure</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"All_DESeq"</span>), </span>
<span id="cb1787-2"><a href="#cb1787-2" aria-hidden="true" tabindex="-1"></a>            All_DESeq, </span>
<span id="cb1787-3"><a href="#cb1787-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">pwidth =</span> <span class="dv">14</span>, </span>
<span id="cb1787-4"><a href="#cb1787-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">pheight =</span> <span class="dv">10</span>,</span>
<span id="cb1787-5"><a href="#cb1787-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 828 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 982 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1126 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1041 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 533 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 519 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 299 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 396 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 828 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 982 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1126 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1041 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 533 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 519 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 299 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 396 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1804"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1804-1"><a href="#cb1804-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"All_DESeq"</span>, <span class="st">".png"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures/All_DESeq.png" class="img-fluid" width="4200"></p>
</div>
</div>
</section>
<section id="plot-labelled-asvs" class="level4">
<h4 class="anchored" data-anchor-id="plot-labelled-asvs">Plot labelled ASVs</h4>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/plot labelled ASVs_df390520d6e5375d4b4395163eedca6e">
<div class="sourceCode cell-code" id="cb1805"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1805-1"><a href="#cb1805-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_combinations &lt;- crossing(Species = c("Epibolus pulchripes", "Glomeris connexa"))</span></span>
<span id="cb1805-2"><a href="#cb1805-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1805-3"><a href="#cb1805-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1805-4"><a href="#cb1805-4" aria-hidden="true" tabindex="-1"></a>params2plot <span class="ot">&lt;-</span> <span class="fu">get_treatment_params</span>(Ps_obj_abs, <span class="fu">c</span>(<span class="st">"Species"</span>,</span>
<span id="cb1805-5"><a href="#cb1805-5" aria-hidden="true" tabindex="-1"></a>                                                  <span class="st">"Label..13C."</span>,</span>
<span id="cb1805-6"><a href="#cb1805-6" aria-hidden="true" tabindex="-1"></a>                                                  <span class="st">"Replicate"</span>),</span>
<span id="cb1805-7"><a href="#cb1805-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"Label..13C. != 'Unlabelled'"</span>)</span>
<span id="cb1805-8"><a href="#cb1805-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1805-9"><a href="#cb1805-9" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&gt;%</span></span>
<span id="cb1805-10"><a href="#cb1805-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(., <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1805-11"><a href="#cb1805-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">taxa_sums</span>(.) <span class="sc">%&gt;%</span> </span>
<span id="cb1805-12"><a href="#cb1805-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_dbl</span>(<span class="sc">~</span>(.x <span class="sc">/</span> <span class="fu">nsamples</span>(Ps_obj_abs))) <span class="sc">%&gt;%</span> </span>
<span id="cb1805-13"><a href="#cb1805-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enframe</span>(<span class="at">name =</span> <span class="st">"ASV"</span>, <span class="at">value =</span> <span class="st">"baseMean"</span>) <span class="ot">-&gt;</span> </span>
<span id="cb1805-14"><a href="#cb1805-14" aria-hidden="true" tabindex="-1"></a>  baseMean</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1814"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1814-1"><a href="#cb1814-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq_res_SIP_byTime_sig_prev_df %&gt;% </span></span>
<span id="cb1814-2"><a href="#cb1814-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(ASV) %&gt;%</span></span>
<span id="cb1814-3"><a href="#cb1814-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   left_join(., baseMean, by = "ASV") %&gt;% # add mean abundance to results table</span></span>
<span id="cb1814-4"><a href="#cb1814-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(tot_mean_abund = mean(baseMean)) %&gt;% </span></span>
<span id="cb1814-5"><a href="#cb1814-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(tot_mean_abund &gt; 0.01) %&gt;% </span></span>
<span id="cb1814-6"><a href="#cb1814-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(ASV, !!sym(Var3)) %&gt;%</span></span>
<span id="cb1814-7"><a href="#cb1814-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(ASV_count = n()) %&gt;% </span></span>
<span id="cb1814-8"><a href="#cb1814-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(ASV_count &gt; prev_thresh) %&gt;%</span></span>
<span id="cb1814-9"><a href="#cb1814-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   ungroup() %&gt;% View</span></span>
<span id="cb1814-10"><a href="#cb1814-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   # add here a filter for only ASVs labelled in at least two time points</span></span>
<span id="cb1814-11"><a href="#cb1814-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(across(all_of(Var1), ~as.numeric(.))) -&gt; # make Replicate numeric</span></span>
<span id="cb1814-12"><a href="#cb1814-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   abund_ASV_2_plot</span></span>
<span id="cb1814-13"><a href="#cb1814-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1814-14"><a href="#cb1814-14" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_sig_prev_con_df <span class="sc">%&gt;%</span> </span>
<span id="cb1814-15"><a href="#cb1814-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ASV) <span class="sc">%&gt;%</span></span>
<span id="cb1814-16"><a href="#cb1814-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., baseMean, <span class="at">by =</span> <span class="st">"ASV"</span>) <span class="sc">%&gt;%</span> <span class="co"># add mean abundance to results table</span></span>
<span id="cb1814-17"><a href="#cb1814-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tot_mean_abund =</span> <span class="fu">mean</span>(baseMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb1814-18"><a href="#cb1814-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(tot_mean_abund <span class="sc">&gt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1814-19"><a href="#cb1814-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(log2FoldChange <span class="sc">&gt;</span> <span class="fl">1.7</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1814-20"><a href="#cb1814-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Significance <span class="sc">==</span> <span class="st">"Pass"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1814-21"><a href="#cb1814-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count_per_group <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1814-22"><a href="#cb1814-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_count</span>(ASV) <span class="sc">%&gt;%</span> </span>
<span id="cb1814-23"><a href="#cb1814-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1814-24"><a href="#cb1814-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ASV, <span class="sc">!!</span><span class="fu">sym</span>(Var3)) <span class="sc">%&gt;%</span></span>
<span id="cb1814-25"><a href="#cb1814-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(ASV_count = n()) %&gt;% </span></span>
<span id="cb1814-26"><a href="#cb1814-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count_per_group <span class="sc">&gt;</span> prev_thresh) <span class="sc">%&gt;%</span></span>
<span id="cb1814-27"><a href="#cb1814-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="ot">-&gt;</span> <span class="co"># make Replicate numeric</span></span>
<span id="cb1814-28"><a href="#cb1814-28" aria-hidden="true" tabindex="-1"></a>  abund_ASV_2_plot</span>
<span id="cb1814-29"><a href="#cb1814-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1814-30"><a href="#cb1814-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Labelled_ASVs &lt;- map(seq(length(Ps_obj_abs_noTime_l)), ~plot_otus_by_density(Ps_obj_abs_noTime_l[[.x]], </span></span>
<span id="cb1814-31"><a href="#cb1814-31" aria-hidden="true" tabindex="-1"></a><span class="co">#                      ASV2plot = filter(abund_ASV_2_plot, Species == params2plot$Species[.x], Replicate == params2plot$Replicate[.x]))) # </span></span>
<span id="cb1814-32"><a href="#cb1814-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1814-33"><a href="#cb1814-33" aria-hidden="true" tabindex="-1"></a>Labelled_ASVs <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">length</span>(Ps_obj_abs_noTime_l)), <span class="sc">~</span><span class="fu">plot_otus_by_density</span>(Ps_obj_abs_noTime_l[[.x]],</span>
<span id="cb1814-34"><a href="#cb1814-34" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ASV2plot =</span> <span class="fu">filter</span>(abund_ASV_2_plot, </span>
<span id="cb1814-35"><a href="#cb1814-35" aria-hidden="true" tabindex="-1"></a>                                       Species <span class="sc">==</span> params2plot<span class="sc">$</span>Species[.x]), </span>
<span id="cb1814-36"><a href="#cb1814-36" aria-hidden="true" tabindex="-1"></a>                     <span class="at">relative =</span> <span class="cn">FALSE</span>, </span>
<span id="cb1814-37"><a href="#cb1814-37" aria-hidden="true" tabindex="-1"></a>                     <span class="at">title =</span> <span class="fu">names</span>(Ps_obj_abs_noTime_l[.x]))) <span class="co">#</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggpomological</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1840"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1840-1"><a href="#cb1840-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">length</span>(Ps_obj_abs_noTime_l)), </span>
<span id="cb1840-2"><a href="#cb1840-2" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">save_figure</span>(<span class="fu">paste0</span>(fig.path, <span class="st">"Labelled_ASVs_"</span>, <span class="fu">paste</span>(params2plot<span class="sc">$</span>Species[.x], params2plot<span class="sc">$</span>Replicate[.x], <span class="at">sep =</span> <span class="st">"_"</span>)), </span>
<span id="cb1840-3"><a href="#cb1840-3" aria-hidden="true" tabindex="-1"></a>                 Labelled_ASVs[[.x]], </span>
<span id="cb1840-4"><a href="#cb1840-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pwidth =</span> <span class="dv">16</span>, </span>
<span id="cb1840-5"><a href="#cb1840-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pheight =</span> <span class="dv">12</span>,</span>
<span id="cb1840-6"><a href="#cb1840-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">dpi =</span> <span class="dv">500</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Epibolus pulchripes_1.svgz"

[[2]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Epibolus pulchripes_2.svgz"

[[3]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Epibolus pulchripes_3.svgz"

[[4]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Glomeris connexa_1.svgz"

[[5]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Glomeris connexa_2.svgz"

[[6]]
[1] "05_SIP_diff_abund_between_DESeq2_figures/Labelled_ASVs_Glomeris connexa_3.svgz"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1842"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1842-1"><a href="#cb1842-1" aria-hidden="true" tabindex="-1"></a>plots2display <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">paste0</span>(fig.path), </span>
<span id="cb1842-2"><a href="#cb1842-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pattern =</span> <span class="st">"^Labelled_ASVs_(.*).png$"</span>,</span>
<span id="cb1842-3"><a href="#cb1842-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1842-4"><a href="#cb1842-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1842-5"><a href="#cb1842-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(plots2display)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Epibolus%20pulchripes_1.png" class="img-fluid" width="4000"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Epibolus%20pulchripes_2.png" class="img-fluid" width="4000"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Epibolus%20pulchripes_3.png" class="img-fluid" width="4000"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Glomeris%20connexa_1.png" class="img-fluid" width="4000"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Glomeris%20connexa_2.png" class="img-fluid" width="4000"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Labelled_ASVs_Glomeris%20connexa_3.png" class="img-fluid" width="4000"></p>
</div>
</div>
</section>
<section id="plot-phylogenetic-trees-with-heatmaps" class="level4">
<h4 class="anchored" data-anchor-id="plot-phylogenetic-trees-with-heatmaps">Plot phylogenetic trees with heatmaps</h4>
<p>Prepare data for plotting</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/Prepare data for tree plotting_336bf13b0089999561259d6c45e9274c">
<div class="sourceCode cell-code" id="cb1843"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1843-1"><a href="#cb1843-1" aria-hidden="true" tabindex="-1"></a><span class="co"># c("Epibolus pulchripes 3 Day", "Epibolus pulchripes 7 Day", "Epibolus pulchripes 14 Day", "Glomeris connexa 3 Day", "Glomeris connexa 7 Day", "Glomeris connexa 14 Day")  -&gt;</span></span>
<span id="cb1843-2"><a href="#cb1843-2" aria-hidden="true" tabindex="-1"></a> <span class="co"># col_order</span></span>
<span id="cb1843-3"><a href="#cb1843-3" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_LFC_sig_prev_con_l <span class="sc">%&gt;%</span> </span>
<span id="cb1843-4"><a href="#cb1843-4" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq_SIP_byTime_l %&gt;% </span></span>
<span id="cb1843-5"><a href="#cb1843-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="sc">~</span><span class="fu">as.data.frame</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb1843-6"><a href="#cb1843-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># map(., ~rownames_to_column(.x, "ASV")) %&gt;% </span></span>
<span id="cb1843-7"><a href="#cb1843-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(., <span class="at">.id =</span> <span class="st">"Comparison"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb1843-8"><a href="#cb1843-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter(str_detect(Comparison, "Labelled")) %&gt;% # remove unlabelled samples [c(-7, -8, -9, -10, -11, -12)]</span></span>
<span id="cb1843-9"><a href="#cb1843-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Significance = ifelse(is.na(paste(vars2test, "- P adj.")) &lt; alpha_thresh &amp; `log2 fold change` &gt; LFC_thresh, "Passed", "Failed")) %&gt;%</span></span>
<span id="cb1843-10"><a href="#cb1843-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Labelled = ifelse(ASV %in% DESeq_res_SIP_byTime_LFC_sig_prev_df$ASV, "Labelled", "Unlabelled")) %&gt;%</span></span>
<span id="cb1843-11"><a href="#cb1843-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ASV) <span class="sc">%&gt;%</span></span>
<span id="cb1843-12"><a href="#cb1843-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., baseMean, <span class="at">by =</span> <span class="st">"ASV"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1843-13"><a href="#cb1843-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1843-14"><a href="#cb1843-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate(., "Comparison" ,c("Species","Day", "Replicate"), sep = " &amp; ") %&gt;% </span></span>
<span id="cb1843-15"><a href="#cb1843-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="st">"Day"</span>),  <span class="sc">~</span><span class="fu">gsub</span>(<span class="st">" d"</span>, <span class="st">""</span>, .))) <span class="sc">%&gt;%</span></span>
<span id="cb1843-16"><a href="#cb1843-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="st">"Day"</span>),  <span class="sc">~</span><span class="fu">as.numeric</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb1843-17"><a href="#cb1843-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Day =</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(Day),</span>
<span id="cb1843-18"><a href="#cb1843-18" aria-hidden="true" tabindex="-1"></a>                          <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"3"</span>, <span class="st">"7"</span>, <span class="st">"14"</span>, <span class="st">"21"</span>)))  <span class="sc">%&gt;%</span> </span>
<span id="cb1843-19"><a href="#cb1843-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Species_Day = paste(Species, Day)) %&gt;%</span></span>
<span id="cb1843-20"><a href="#cb1843-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(across(Species_Day, ~factor(., levels = col_order))) %&gt;% </span></span>
<span id="cb1843-21"><a href="#cb1843-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">factor</span>(<span class="fu">paste0</span>(Species),</span>
<span id="cb1843-22"><a href="#cb1843-22" aria-hidden="true" tabindex="-1"></a>                          <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Epibolus pulchripes"</span>, <span class="st">"Glomeris connexa"</span>)))  <span class="sc">%&gt;%</span> </span>
<span id="cb1843-23"><a href="#cb1843-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1843-24"><a href="#cb1843-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(Species = paste(Species)) %&gt;% </span></span>
<span id="cb1843-25"><a href="#cb1843-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="ot">-&gt;</span></span>
<span id="cb1843-26"><a href="#cb1843-26" aria-hidden="true" tabindex="-1"></a>  DESeq_res_SIP_byTime_all_df </span>
<span id="cb1843-27"><a href="#cb1843-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1843-28"><a href="#cb1843-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1843-29"><a href="#cb1843-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarise number of labelled and unlabelled ASVs</span></span>
<span id="cb1843-30"><a href="#cb1843-30" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_all_df <span class="sc">%&gt;%</span> </span>
<span id="cb1843-31"><a href="#cb1843-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Significance) <span class="sc">%&gt;%</span> </span>
<span id="cb1843-32"><a href="#cb1843-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 2
  Significance     n
  &lt;chr&gt;        &lt;int&gt;
1 Fail          5722
2 Pass          1796</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1845"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1845-1"><a href="#cb1845-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove taxa with NA from DESeq analysis</span></span>
<span id="cb1845-2"><a href="#cb1845-2" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_all_df <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb1845-3"><a href="#cb1845-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Significance)) <span class="co"># %&gt;% </span></span>
<span id="cb1845-4"><a href="#cb1845-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pull(Labelled) -&gt; </span></span>
<span id="cb1845-5"><a href="#cb1845-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># bad_seqs</span></span>
<span id="cb1845-6"><a href="#cb1845-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1845-7"><a href="#cb1845-7" aria-hidden="true" tabindex="-1"></a><span class="co"># remove NA taxa from PS obj</span></span>
<span id="cb1845-8"><a href="#cb1845-8" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&gt;%</span> </span>
<span id="cb1845-9"><a href="#cb1845-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prune_taxa</span>(<span class="fu">setdiff</span>(<span class="fu">taxa_names</span>(Ps_obj_abs), <span class="st">""</span>), .) <span class="sc">%&gt;%</span> <span class="co"># outlier</span></span>
<span id="cb1845-10"><a href="#cb1845-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prune_taxa</span>(DESeq_res_SIP_byTime_all_df<span class="sc">$</span>ASV, .) <span class="ot">-&gt;</span></span>
<span id="cb1845-11"><a href="#cb1845-11" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs4tree_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1850"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1850-1"><a href="#cb1850-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove long name</span></span>
<span id="cb1850-2"><a href="#cb1850-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tax_table(Ps_obj_abs4tree_plot)[, "Order"] %&lt;&gt;%  str_replace_all(., "Gammaproteobacteria Incertae Sedis", "Incertae Sedis")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot heatmap to show the difference between the replicates</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/Plot heatmaps_79ce4f65da4e8dee20126c11ec978155">
<div class="sourceCode cell-code" id="cb1851"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1851-1"><a href="#cb1851-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get top Epi ASVs</span></span>
<span id="cb1851-2"><a href="#cb1851-2" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb1851-3"><a href="#cb1851-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">==</span> <span class="st">"Epibolus pulchripes"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-4"><a href="#cb1851-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(baseMean)) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-5"><a href="#cb1851-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ASV) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-6"><a href="#cb1851-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">200</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-7"><a href="#cb1851-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ASV) <span class="ot">-&gt;</span></span>
<span id="cb1851-8"><a href="#cb1851-8" aria-hidden="true" tabindex="-1"></a>  top_Epi_ASVs</span>
<span id="cb1851-9"><a href="#cb1851-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1851-10"><a href="#cb1851-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make data frame to plot</span></span>
<span id="cb1851-11"><a href="#cb1851-11" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb1851-12"><a href="#cb1851-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">==</span> <span class="st">"Epibolus pulchripes"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-13"><a href="#cb1851-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> top_Epi_ASVs) <span class="sc">%&gt;%</span> </span>
<span id="cb1851-14"><a href="#cb1851-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(Day, <span class="sc">~</span><span class="fu">factor</span>(., <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"3 d"</span>, <span class="st">"7 d"</span>, <span class="st">"14 d"</span>, <span class="st">"21 d"</span>)))) <span class="ot">-&gt;</span></span>
<span id="cb1851-15"><a href="#cb1851-15" aria-hidden="true" tabindex="-1"></a>  DESeq_res_SIP_byTime_LFC_sig_Epi_top_df</span>
<span id="cb1851-16"><a href="#cb1851-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1851-17"><a href="#cb1851-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_reps_heatmap</span>(DESeq_res_SIP_byTime_LFC_sig_Epi_top_df, <span class="at">HM_title =</span> <span class="st">"*Epibolus pulchirpes*"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/Plot%20heatmaps-1.png" class="img-fluid" width="960"></p>
</div>
<div class="sourceCode cell-code" id="cb1853"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1853-1"><a href="#cb1853-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get top Glo ASVs</span></span>
<span id="cb1853-2"><a href="#cb1853-2" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb1853-3"><a href="#cb1853-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">==</span> <span class="st">"Glomeris connexa"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1853-4"><a href="#cb1853-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(baseMean)) <span class="sc">%&gt;%</span>  </span>
<span id="cb1853-5"><a href="#cb1853-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ASV) <span class="sc">%&gt;%</span> </span>
<span id="cb1853-6"><a href="#cb1853-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">200</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1853-7"><a href="#cb1853-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ASV) <span class="ot">-&gt;</span></span>
<span id="cb1853-8"><a href="#cb1853-8" aria-hidden="true" tabindex="-1"></a>  top_Glo_ASVs</span>
<span id="cb1853-9"><a href="#cb1853-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1853-10"><a href="#cb1853-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make data frame to plot</span></span>
<span id="cb1853-11"><a href="#cb1853-11" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_sig_prev_df <span class="sc">%&gt;%</span> </span>
<span id="cb1853-12"><a href="#cb1853-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">==</span> <span class="st">"Glomeris connexa"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1853-13"><a href="#cb1853-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> top_Glo_ASVs) <span class="sc">%&gt;%</span> </span>
<span id="cb1853-14"><a href="#cb1853-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(Day, <span class="sc">~</span><span class="fu">factor</span>(., <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"3 d"</span>, <span class="st">"7 d"</span>, <span class="st">"14 d"</span>, <span class="st">"21 d"</span>)))) <span class="ot">-&gt;</span></span>
<span id="cb1853-15"><a href="#cb1853-15" aria-hidden="true" tabindex="-1"></a>  DESeq_res_SIP_byTime_LFC_sig_Glo_top_df</span>
<span id="cb1853-16"><a href="#cb1853-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1853-17"><a href="#cb1853-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_reps_heatmap</span>(DESeq_res_SIP_byTime_LFC_sig_Glo_top_df, <span class="at">HM_title =</span> <span class="st">"*Glomeris connexa*"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="37_SIP_diff_abund_between_DESeq2_files/figure-html/Plot%20heatmaps-2.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Now plot the phylogenetic trees with heatmaps</p>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/Plot trees with heatmaps_03cb945e5c3b653b5a5722ff5fc311d2">
<div class="sourceCode cell-code" id="cb1854"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1854-1"><a href="#cb1854-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define plot grid</span></span>
<span id="cb1854-2"><a href="#cb1854-2" aria-hidden="true" tabindex="-1"></a>taxa2plot <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">rank =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Class"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"Phylum"</span>, <span class="dv">4</span>)), </span>
<span id="cb1854-3"><a href="#cb1854-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">subrank =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Order"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"Class"</span>, <span class="dv">4</span>)), </span>
<span id="cb1854-4"><a href="#cb1854-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Taxa2plot =</span> <span class="fu">c</span>(<span class="st">"Actinobacteria"</span>,</span>
<span id="cb1854-5"><a href="#cb1854-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Alphaproteobacteria"</span>, </span>
<span id="cb1854-6"><a href="#cb1854-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Gammaproteobacteria"</span>,</span>
<span id="cb1854-7"><a href="#cb1854-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Bacteroidia"</span>,</span>
<span id="cb1854-8"><a href="#cb1854-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Desulfobacterota"</span>,</span>
<span id="cb1854-9"><a href="#cb1854-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Bacillota"</span>,</span>
<span id="cb1854-10"><a href="#cb1854-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Planctomycetota"</span>,</span>
<span id="cb1854-11"><a href="#cb1854-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Verrucomicrobiota"</span>),</span>
<span id="cb1854-12"><a href="#cb1854-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">l_rows =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>),</span>
<span id="cb1854-13"><a href="#cb1854-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pwidth =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>), </span>
<span id="cb1854-14"><a href="#cb1854-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pheight =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">8</span>, <span class="dv">6</span>, <span class="dv">10</span>, <span class="dv">6</span>, <span class="dv">6</span>))</span>
<span id="cb1854-15"><a href="#cb1854-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1854-16"><a href="#cb1854-16" aria-hidden="true" tabindex="-1"></a>tree_p_l <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(taxa2plot)), </span>
<span id="cb1854-17"><a href="#cb1854-17" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span><span class="fu">wrap_tree_heatmap</span>(Ps_obj_abs4tree_plot,</span>
<span id="cb1854-18"><a href="#cb1854-18" aria-hidden="true" tabindex="-1"></a>                                   DESeq_res_SIP_byTime_all_df,</span>
<span id="cb1854-19"><a href="#cb1854-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">rank =</span> taxa2plot<span class="sc">$</span>rank[.x],</span>
<span id="cb1854-20"><a href="#cb1854-20" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">subrank =</span> taxa2plot<span class="sc">$</span>subrank[.x],</span>
<span id="cb1854-21"><a href="#cb1854-21" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">Taxa2plot =</span> taxa2plot<span class="sc">$</span>Taxa2plot[.x],</span>
<span id="cb1854-22"><a href="#cb1854-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">x =</span> <span class="st">"Day"</span>,</span>
<span id="cb1854-23"><a href="#cb1854-23" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">facet_names =</span> <span class="st">"Species"</span>,</span>
<span id="cb1854-24"><a href="#cb1854-24" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ntaxa =</span> <span class="dv">400</span>,</span>
<span id="cb1854-25"><a href="#cb1854-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">l_rows =</span> taxa2plot<span class="sc">$</span>l_rows[.x],</span>
<span id="cb1854-26"><a href="#cb1854-26" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pwidth =</span> taxa2plot<span class="sc">$</span>pwidth[.x],</span>
<span id="cb1854-27"><a href="#cb1854-27" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pheight =</span> taxa2plot<span class="sc">$</span>pheight[.x])) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: RColorBrewer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1936"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1936-1"><a href="#cb1936-1" aria-hidden="true" tabindex="-1"></a>DESeq_res_SIP_byTime_all_df <span class="sc">%&gt;%</span> </span>
<span id="cb1936-2"><a href="#cb1936-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">rownames_to_column</span>(<span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(Ps_obj_abs4tree_plot)), <span class="st">"ASV"</span>), <span class="at">by =</span> <span class="st">"ASV"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1936-3"><a href="#cb1936-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(., <span class="at">file =</span> <span class="st">"DESeq2_byTime_a-0.051_wTax.tsv"</span>)</span>
<span id="cb1936-4"><a href="#cb1936-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1936-5"><a href="#cb1936-5" aria-hidden="true" tabindex="-1"></a>trees2display <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">paste0</span>(fig.path), </span>
<span id="cb1936-6"><a href="#cb1936-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pattern =</span> <span class="st">"^Tree_HM_(.*).png$"</span>,</span>
<span id="cb1936-7"><a href="#cb1936-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1936-8"><a href="#cb1936-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1936-9"><a href="#cb1936-9" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(trees2display)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Actinobacteria.png" class="img-fluid" width="1800"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Alphaproteobacteria.png" class="img-fluid" width="1800"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Bacillota.png" class="img-fluid" width="1200"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Bacteroidia.png" class="img-fluid" width="1800"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Desulfobacterota.png" class="img-fluid" width="1200"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Gammaproteobacteria.png" class="img-fluid" width="1800"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Planctomycetota.png" class="img-fluid" width="1200"></p>
</div>
<div class="cell-output-display">
<p><img src="05_SIP_diff_abund_between_DESeq2_figures//Tree_HM_Verrucomicrobiota.png" class="img-fluid" width="1200"></p>
</div>
<div class="sourceCode cell-code" id="cb1937"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1937-1"><a href="#cb1937-1" aria-hidden="true" tabindex="-1"></a><span class="co"># all_trees &lt;- ((tree_p_l[[1]] | tree_p_l[[2]] + guides(fill = FALSE) | tree_p_l[[3]] + guides(fill = FALSE) | tree_p_l[[4]] + guides(fill = FALSE)) / (tree_p_l[[5]] | plot_spacer())) + plot_layout(heights = c(2, 1))</span></span>
<span id="cb1937-2"><a href="#cb1937-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1937-3"><a href="#cb1937-3" aria-hidden="true" tabindex="-1"></a><span class="co"># save_figure(paste0(fig.path, "all_trees"), </span></span>
<span id="cb1937-4"><a href="#cb1937-4" aria-hidden="true" tabindex="-1"></a><span class="co">#             all_trees, </span></span>
<span id="cb1937-5"><a href="#cb1937-5" aria-hidden="true" tabindex="-1"></a><span class="co">#             pwidth = 16, </span></span>
<span id="cb1937-6"><a href="#cb1937-6" aria-hidden="true" tabindex="-1"></a><span class="co">#             pheight = 18,</span></span>
<span id="cb1937-7"><a href="#cb1937-7" aria-hidden="true" tabindex="-1"></a><span class="co">#             dpi = 900)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="how-abundant-were-the-labelled-asvs" class="level4">
<h4 class="anchored" data-anchor-id="how-abundant-were-the-labelled-asvs">How abundant were the labelled ASVs?</h4>
<div class="cell" data-hash="37_SIP_diff_abund_between_DESeq2_cache/html/labelled ASV abundance_4f74e123a915e76bc54f3bda4fe92221">
<div class="sourceCode cell-code" id="cb1938"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1938-1"><a href="#cb1938-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs <span class="sc">%&gt;%</span> </span>
<span id="cb1938-2"><a href="#cb1938-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="sc">/</span><span class="fu">sum</span>(x)}) <span class="sc">%&gt;%</span> </span>
<span id="cb1938-3"><a href="#cb1938-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prune_taxa</span>(<span class="fu">filter</span>(DESeq_res_SIP_byTime_LFC_sig_prev_con_df, </span>
<span id="cb1938-4"><a href="#cb1938-4" aria-hidden="true" tabindex="-1"></a>                    Species <span class="sc">==</span> <span class="st">"Epibolus pulchripes"</span> <span class="sc">&amp;</span> Significance <span class="sc">==</span> <span class="st">"Pass"</span>)<span class="sc">$</span>ASV, .) <span class="sc">%&gt;%</span></span>
<span id="cb1938-5"><a href="#cb1938-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_samples</span>(Species <span class="sc">==</span> <span class="st">"Epibolus pulchripes"</span> <span class="sc">&amp;</span> Group <span class="sc">==</span> <span class="st">"LabelledHeavy"</span>) <span class="ot">-&gt;</span> </span>
<span id="cb1938-6"><a href="#cb1938-6" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_EP</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1957"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1957-1"><a href="#cb1957-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_EP <span class="sc">%&gt;%</span> </span>
<span id="cb1957-2"><a href="#cb1957-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">otu_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1957-3"><a href="#cb1957-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as</span>(., <span class="st">"matrix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-4"><a href="#cb1957-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowSums</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1957-5"><a href="#cb1957-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">Tot_rel_abund =</span> . <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-6"><a href="#cb1957-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"sample_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-7"><a href="#cb1957-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(<span class="fu">rownames_to_column</span>(<span class="fu">get_variable</span>(Ps_obj_abs_EP), <span class="st">"sample_ID"</span>), ., <span class="at">by =</span> <span class="st">"sample_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-8"><a href="#cb1957-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Label..13C.</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Labelled"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-9"><a href="#cb1957-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Species, Day, Tot_rel_abund) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-10"><a href="#cb1957-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Species, Day) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-11"><a href="#cb1957-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_if</span>(is.numeric, mean) <span class="ot">-&gt;</span></span>
<span id="cb1957-12"><a href="#cb1957-12" aria-hidden="true" tabindex="-1"></a>  EP</span>
<span id="cb1957-13"><a href="#cb1957-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1957-14"><a href="#cb1957-14" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs <span class="sc">%&gt;%</span> </span>
<span id="cb1957-15"><a href="#cb1957-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="sc">/</span><span class="fu">sum</span>(x)}) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-16"><a href="#cb1957-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prune_taxa</span>(<span class="fu">filter</span>(DESeq_res_SIP_byTime_LFC_sig_prev_con_df, </span>
<span id="cb1957-17"><a href="#cb1957-17" aria-hidden="true" tabindex="-1"></a>                    Species <span class="sc">==</span> <span class="st">"Glomeris connexa"</span> <span class="sc">&amp;</span> Significance <span class="sc">==</span> <span class="st">"Pass"</span>)<span class="sc">$</span>ASV, .) <span class="sc">%&gt;%</span> </span>
<span id="cb1957-18"><a href="#cb1957-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_samples</span>(Species <span class="sc">==</span> <span class="st">"Glomeris connexa"</span>) <span class="ot">-&gt;</span> </span>
<span id="cb1957-19"><a href="#cb1957-19" aria-hidden="true" tabindex="-1"></a>  Ps_obj_abs_GC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb1975"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1975-1"><a href="#cb1975-1" aria-hidden="true" tabindex="-1"></a>Ps_obj_abs_GC <span class="sc">%&gt;%</span> </span>
<span id="cb1975-2"><a href="#cb1975-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">otu_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1975-3"><a href="#cb1975-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as</span>(., <span class="st">"matrix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-4"><a href="#cb1975-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowSums</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1975-5"><a href="#cb1975-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">Tot_rel_abund =</span> . <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-6"><a href="#cb1975-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"sample_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-7"><a href="#cb1975-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(<span class="fu">rownames_to_column</span>(<span class="fu">get_variable</span>(Ps_obj_abs_GC), <span class="st">"sample_ID"</span>), ., <span class="at">by =</span> <span class="st">"sample_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-8"><a href="#cb1975-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Label..13C.</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Labelled"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-9"><a href="#cb1975-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Species, Day, Tot_rel_abund) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-10"><a href="#cb1975-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Species, Day) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-11"><a href="#cb1975-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_if</span>(is.numeric, mean) <span class="ot">-&gt;</span></span>
<span id="cb1975-12"><a href="#cb1975-12" aria-hidden="true" tabindex="-1"></a>  GC</span>
<span id="cb1975-13"><a href="#cb1975-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1975-14"><a href="#cb1975-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1975-15"><a href="#cb1975-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(EP, GC) <span class="sc">%&gt;%</span> </span>
<span id="cb1975-16"><a href="#cb1975-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(., <span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1975-17"><a href="#cb1975-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>, <span class="st">"condensed"</span>, <span class="st">"responsive"</span>), </span>
<span id="cb1975-18"><a href="#cb1975-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">full_width =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Species </th>
   <th style="text-align:left;"> Day </th>
   <th style="text-align:right;"> Tot_rel_abund </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Epibolus pulchripes </td>
   <td style="text-align:left;"> 3 </td>
   <td style="text-align:right;"> 71.873 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Epibolus pulchripes </td>
   <td style="text-align:left;"> 7 </td>
   <td style="text-align:right;"> 66.807 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Epibolus pulchripes </td>
   <td style="text-align:left;"> 14 </td>
   <td style="text-align:right;"> 70.715 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Epibolus pulchripes </td>
   <td style="text-align:left;"> 21 </td>
   <td style="text-align:right;"> 61.494 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Glomeris connexa </td>
   <td style="text-align:left;"> 3 </td>
   <td style="text-align:right;"> 83.780 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Glomeris connexa </td>
   <td style="text-align:left;"> 7 </td>
   <td style="text-align:right;"> 83.163 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Glomeris connexa </td>
   <td style="text-align:left;"> 14 </td>
   <td style="text-align:right;"> 76.054 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Glomeris connexa </td>
   <td style="text-align:left;"> 21 </td>
   <td style="text-align:right;"> 68.919 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb1976"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1976-1"><a href="#cb1976-1" aria-hidden="true" tabindex="-1"></a>sessioninfo<span class="sc">::</span><span class="fu">session_info</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1976-2"><a href="#cb1976-2" aria-hidden="true" tabindex="-1"></a>  details<span class="sc">::</span><span class="fu">details</span>(</span>
<span id="cb1976-3"><a href="#cb1976-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">summary =</span> <span class="st">'Current session info'</span>,</span>
<span id="cb1976-4"><a href="#cb1976-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">open    =</span> <span class="cn">TRUE</span></span>
<span id="cb1976-5"><a href="#cb1976-5" aria-hidden="true" tabindex="-1"></a> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<details open="">
<summary>
<span title="Click to Expand"> Current session info </span>
</summary>
<div class="sourceCode" id="cb1977"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1977-1"><a href="#cb1977-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1977-2"><a href="#cb1977-2" aria-hidden="true" tabindex="-1"></a>─ Session info ───────────────────────────────────────────────────────────────</span>
<span id="cb1977-3"><a href="#cb1977-3" aria-hidden="true" tabindex="-1"></a> setting  value</span>
<span id="cb1977-4"><a href="#cb1977-4" aria-hidden="true" tabindex="-1"></a> version  R version <span class="dv">4</span>.<span class="fl">4.0</span> (<span class="dv">2024-04-24</span>)</span>
<span id="cb1977-5"><a href="#cb1977-5" aria-hidden="true" tabindex="-1"></a> os       Ubuntu <span class="dv">22</span>.<span class="fl">04.4</span> LTS</span>
<span id="cb1977-6"><a href="#cb1977-6" aria-hidden="true" tabindex="-1"></a> system   x86_64, linux<span class="sc">-</span>gnu</span>
<span id="cb1977-7"><a href="#cb1977-7" aria-hidden="true" tabindex="-1"></a> ui       X11</span>
<span id="cb1977-8"><a href="#cb1977-8" aria-hidden="true" tabindex="-1"></a> <span class="fu">language</span> (EN)</span>
<span id="cb1977-9"><a href="#cb1977-9" aria-hidden="true" tabindex="-1"></a> collate  en_US.UTF<span class="dv">-8</span></span>
<span id="cb1977-10"><a href="#cb1977-10" aria-hidden="true" tabindex="-1"></a> ctype    en_US.UTF<span class="dv">-8</span></span>
<span id="cb1977-11"><a href="#cb1977-11" aria-hidden="true" tabindex="-1"></a> tz       Europe<span class="sc">/</span>Prague</span>
<span id="cb1977-12"><a href="#cb1977-12" aria-hidden="true" tabindex="-1"></a> date     <span class="dv">2024-06-17</span></span>
<span id="cb1977-13"><a href="#cb1977-13" aria-hidden="true" tabindex="-1"></a> pandoc   <span class="dv">2</span>.<span class="fl">19.2</span> <span class="sc">@</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>rstudio<span class="sc">-</span>server<span class="sc">/</span>bin<span class="sc">/</span>quarto<span class="sc">/</span>bin<span class="sc">/</span>tools<span class="sc">/</span> (via rmarkdown)</span>
<span id="cb1977-14"><a href="#cb1977-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1977-15"><a href="#cb1977-15" aria-hidden="true" tabindex="-1"></a>─ Packages ───────────────────────────────────────────────────────────────────</span>
<span id="cb1977-16"><a href="#cb1977-16" aria-hidden="true" tabindex="-1"></a> package              <span class="sc">*</span> version    <span class="fu">date</span> (UTC) lib source</span>
<span id="cb1977-17"><a href="#cb1977-17" aria-hidden="true" tabindex="-1"></a> abind                  <span class="fl">1.4</span><span class="dv">-5</span>      <span class="dv">2016-07-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-18"><a href="#cb1977-18" aria-hidden="true" tabindex="-1"></a> ade4                   <span class="fl">1.7</span><span class="dv">-22</span>     <span class="dv">2023-02-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-19"><a href="#cb1977-19" aria-hidden="true" tabindex="-1"></a> ape                    <span class="fl">5.8</span>        <span class="dv">2024-04-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-20"><a href="#cb1977-20" aria-hidden="true" tabindex="-1"></a> aplot                  <span class="dv">0</span>.<span class="fl">2.3</span>      <span class="dv">2024-06-17</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-21"><a href="#cb1977-21" aria-hidden="true" tabindex="-1"></a> ashr                 <span class="sc">*</span> <span class="fl">2.2</span><span class="dv">-63</span>     <span class="dv">2023-08-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-22"><a href="#cb1977-22" aria-hidden="true" tabindex="-1"></a> backports              <span class="dv">1</span>.<span class="fl">5.0</span>      <span class="dv">2024-05-23</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-23"><a href="#cb1977-23" aria-hidden="true" tabindex="-1"></a> Biobase              <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">60.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-24"><a href="#cb1977-24" aria-hidden="true" tabindex="-1"></a> BiocGenerics         <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">46.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-25"><a href="#cb1977-25" aria-hidden="true" tabindex="-1"></a> BiocParallel           <span class="dv">1</span>.<span class="fl">34.2</span>     <span class="dv">2023-05-22</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-26"><a href="#cb1977-26" aria-hidden="true" tabindex="-1"></a> biomformat             <span class="dv">1</span>.<span class="fl">28.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-27"><a href="#cb1977-27" aria-hidden="true" tabindex="-1"></a> Biostrings           <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">68.1</span>     <span class="dv">2023-05-16</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-28"><a href="#cb1977-28" aria-hidden="true" tabindex="-1"></a> bitops                 <span class="fl">1.0</span><span class="dv">-7</span>      <span class="dv">2021-04-24</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.3</span>)</span>
<span id="cb1977-29"><a href="#cb1977-29" aria-hidden="true" tabindex="-1"></a> broom                  <span class="dv">1</span>.<span class="fl">0.6</span>      <span class="dv">2024-05-17</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-30"><a href="#cb1977-30" aria-hidden="true" tabindex="-1"></a> cachem                 <span class="dv">1</span>.<span class="fl">1.0</span>      <span class="dv">2024-05-16</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-31"><a href="#cb1977-31" aria-hidden="true" tabindex="-1"></a> car                    <span class="fl">3.1</span><span class="dv">-2</span>      <span class="dv">2023-03-30</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-32"><a href="#cb1977-32" aria-hidden="true" tabindex="-1"></a> carData                <span class="fl">3.0</span><span class="dv">-5</span>      <span class="dv">2022-01-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-33"><a href="#cb1977-33" aria-hidden="true" tabindex="-1"></a> cli                    <span class="dv">3</span>.<span class="fl">6.2</span>      <span class="dv">2023-12-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-34"><a href="#cb1977-34" aria-hidden="true" tabindex="-1"></a> clipr                  <span class="dv">0</span>.<span class="fl">8.0</span>      <span class="dv">2022-02-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-35"><a href="#cb1977-35" aria-hidden="true" tabindex="-1"></a> cluster                <span class="dv">2</span>.<span class="fl">1.6</span>      <span class="dv">2023-12-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-36"><a href="#cb1977-36" aria-hidden="true" tabindex="-1"></a> codetools              <span class="fl">0.2</span><span class="dv">-20</span>     <span class="dv">2024-03-31</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-37"><a href="#cb1977-37" aria-hidden="true" tabindex="-1"></a> colorspace             <span class="fl">2.1</span><span class="dv">-0</span>      <span class="dv">2023-01-23</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-38"><a href="#cb1977-38" aria-hidden="true" tabindex="-1"></a> commonmark             <span class="dv">1</span>.<span class="fl">9.1</span>      <span class="dv">2024-01-30</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-39"><a href="#cb1977-39" aria-hidden="true" tabindex="-1"></a> cowplot                <span class="dv">1</span>.<span class="fl">1.3</span>      <span class="dv">2024-01-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-40"><a href="#cb1977-40" aria-hidden="true" tabindex="-1"></a> crayon                 <span class="dv">1</span>.<span class="fl">5.2</span>      <span class="dv">2022-09-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.1</span>)</span>
<span id="cb1977-41"><a href="#cb1977-41" aria-hidden="true" tabindex="-1"></a> data.table             <span class="dv">1</span>.<span class="fl">15.4</span>     <span class="dv">2024-03-30</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-42"><a href="#cb1977-42" aria-hidden="true" tabindex="-1"></a> DelayedArray           <span class="dv">0</span>.<span class="fl">26.7</span>     <span class="dv">2023-07-28</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-43"><a href="#cb1977-43" aria-hidden="true" tabindex="-1"></a> desc                   <span class="dv">1</span>.<span class="fl">4.3</span>      <span class="dv">2023-12-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-44"><a href="#cb1977-44" aria-hidden="true" tabindex="-1"></a> DESeq2               <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">40.2</span>     <span class="dv">2023-06-23</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-45"><a href="#cb1977-45" aria-hidden="true" tabindex="-1"></a> details                <span class="dv">0</span>.<span class="fl">3.0</span>      <span class="dv">2022-03-27</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.3</span>)</span>
<span id="cb1977-46"><a href="#cb1977-46" aria-hidden="true" tabindex="-1"></a> digest                 <span class="dv">0</span>.<span class="fl">6.35</span>     <span class="dv">2024-03-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-47"><a href="#cb1977-47" aria-hidden="true" tabindex="-1"></a> dplyr                <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">1.4</span>      <span class="dv">2023-11-17</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-48"><a href="#cb1977-48" aria-hidden="true" tabindex="-1"></a> evaluate               <span class="dv">0</span>.<span class="fl">24.0</span>     <span class="dv">2024-06-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-49"><a href="#cb1977-49" aria-hidden="true" tabindex="-1"></a> extrafont            <span class="sc">*</span> <span class="fl">0.19</span>       <span class="dv">2023-01-18</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-50"><a href="#cb1977-50" aria-hidden="true" tabindex="-1"></a> extrafontdb            <span class="fl">1.0</span>        <span class="dv">2012-06-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-51"><a href="#cb1977-51" aria-hidden="true" tabindex="-1"></a> fansi                  <span class="dv">1</span>.<span class="fl">0.6</span>      <span class="dv">2023-12-08</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-52"><a href="#cb1977-52" aria-hidden="true" tabindex="-1"></a> farver                 <span class="dv">2</span>.<span class="fl">1.2</span>      <span class="dv">2024-05-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-53"><a href="#cb1977-53" aria-hidden="true" tabindex="-1"></a> fastmap                <span class="dv">1</span>.<span class="fl">2.0</span>      <span class="dv">2024-05-15</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-54"><a href="#cb1977-54" aria-hidden="true" tabindex="-1"></a> forcats              <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">0.0</span>      <span class="dv">2023-01-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-55"><a href="#cb1977-55" aria-hidden="true" tabindex="-1"></a> foreach                <span class="dv">1</span>.<span class="fl">5.2</span>      <span class="dv">2022-02-02</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-56"><a href="#cb1977-56" aria-hidden="true" tabindex="-1"></a> fs                     <span class="dv">1</span>.<span class="fl">6.4</span>      <span class="dv">2024-04-25</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-57"><a href="#cb1977-57" aria-hidden="true" tabindex="-1"></a> generics               <span class="dv">0</span>.<span class="fl">1.3</span>      <span class="dv">2022-07-05</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.0</span>)</span>
<span id="cb1977-58"><a href="#cb1977-58" aria-hidden="true" tabindex="-1"></a> GenomeInfoDb         <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">36.2</span>     <span class="dv">2023-08-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-59"><a href="#cb1977-59" aria-hidden="true" tabindex="-1"></a> GenomeInfoDbData       <span class="dv">1</span>.<span class="fl">2.10</span>     <span class="dv">2023-05-31</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-60"><a href="#cb1977-60" aria-hidden="true" tabindex="-1"></a> GenomicRanges        <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">52.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-61"><a href="#cb1977-61" aria-hidden="true" tabindex="-1"></a> ggfun                  <span class="dv">0</span>.<span class="fl">1.5</span>      <span class="dv">2024-05-28</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-62"><a href="#cb1977-62" aria-hidden="true" tabindex="-1"></a> ggplot2              <span class="sc">*</span> <span class="dv">3</span>.<span class="fl">5.1</span>      <span class="dv">2024-04-23</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-63"><a href="#cb1977-63" aria-hidden="true" tabindex="-1"></a> ggplotify              <span class="dv">0</span>.<span class="fl">1.2</span>      <span class="dv">2023-08-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-64"><a href="#cb1977-64" aria-hidden="true" tabindex="-1"></a> ggpomological        <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">1.2</span>      <span class="dv">2020-08-13</span> [<span class="dv">1</span>] <span class="fu">Github</span> (gadenbuie<span class="sc">/</span>ggpomological<span class="sc">@</span>69f3815)</span>
<span id="cb1977-65"><a href="#cb1977-65" aria-hidden="true" tabindex="-1"></a> ggpubr               <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">6.0</span>      <span class="dv">2023-02-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-66"><a href="#cb1977-66" aria-hidden="true" tabindex="-1"></a> ggrepel              <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">9.5</span>      <span class="dv">2024-01-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-67"><a href="#cb1977-67" aria-hidden="true" tabindex="-1"></a> ggsci                <span class="sc">*</span> <span class="dv">3</span>.<span class="fl">1.0</span>      <span class="dv">2024-05-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-68"><a href="#cb1977-68" aria-hidden="true" tabindex="-1"></a> ggsignif               <span class="dv">0</span>.<span class="fl">6.4</span>      <span class="dv">2022-10-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-69"><a href="#cb1977-69" aria-hidden="true" tabindex="-1"></a> ggtext               <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">1.2</span>      <span class="dv">2022-09-16</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.1</span>)</span>
<span id="cb1977-70"><a href="#cb1977-70" aria-hidden="true" tabindex="-1"></a> ggtree               <span class="sc">*</span> <span class="dv">3</span>.<span class="fl">8.2</span>      <span class="dv">2023-07-24</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-71"><a href="#cb1977-71" aria-hidden="true" tabindex="-1"></a> glue                 <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">7.0</span>      <span class="dv">2024-01-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-72"><a href="#cb1977-72" aria-hidden="true" tabindex="-1"></a> gridExtra              <span class="fl">2.3</span>        <span class="dv">2017-09-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-73"><a href="#cb1977-73" aria-hidden="true" tabindex="-1"></a> gridGraphics           <span class="fl">0.5</span><span class="dv">-1</span>      <span class="dv">2020-12-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-74"><a href="#cb1977-74" aria-hidden="true" tabindex="-1"></a> gridtext               <span class="dv">0</span>.<span class="fl">1.5</span>      <span class="dv">2022-09-16</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.1</span>)</span>
<span id="cb1977-75"><a href="#cb1977-75" aria-hidden="true" tabindex="-1"></a> gtable                 <span class="dv">0</span>.<span class="fl">3.5</span>      <span class="dv">2024-04-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-76"><a href="#cb1977-76" aria-hidden="true" tabindex="-1"></a> hms                    <span class="dv">1</span>.<span class="fl">1.3</span>      <span class="dv">2023-03-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-77"><a href="#cb1977-77" aria-hidden="true" tabindex="-1"></a> htmltools              <span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>    <span class="dv">2024-04-04</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-78"><a href="#cb1977-78" aria-hidden="true" tabindex="-1"></a> htmlwidgets            <span class="dv">1</span>.<span class="fl">6.4</span>      <span class="dv">2023-12-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-79"><a href="#cb1977-79" aria-hidden="true" tabindex="-1"></a> HTSSIP               <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">4.1</span>      <span class="dv">2021-01-15</span> [<span class="dv">1</span>] <span class="fu">Github</span> (buckleylab<span class="sc">/</span>HTSSIP<span class="sc">@</span>29ec56b)</span>
<span id="cb1977-80"><a href="#cb1977-80" aria-hidden="true" tabindex="-1"></a> httr                   <span class="dv">1</span>.<span class="fl">4.7</span>      <span class="dv">2023-08-15</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-81"><a href="#cb1977-81" aria-hidden="true" tabindex="-1"></a> igraph                 <span class="dv">2</span>.<span class="fl">0.3</span>      <span class="dv">2024-03-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-82"><a href="#cb1977-82" aria-hidden="true" tabindex="-1"></a> invgamma               <span class="fl">1.1</span>        <span class="dv">2017-05-07</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-83"><a href="#cb1977-83" aria-hidden="true" tabindex="-1"></a> IRanges              <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">34.1</span>     <span class="dv">2023-06-22</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-84"><a href="#cb1977-84" aria-hidden="true" tabindex="-1"></a> irlba                  <span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>    <span class="dv">2022-10-03</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.1</span>)</span>
<span id="cb1977-85"><a href="#cb1977-85" aria-hidden="true" tabindex="-1"></a> iterators              <span class="dv">1</span>.<span class="fl">0.14</span>     <span class="dv">2022-02-05</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-86"><a href="#cb1977-86" aria-hidden="true" tabindex="-1"></a> jsonlite               <span class="dv">1</span>.<span class="fl">8.8</span>      <span class="dv">2023-12-04</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-87"><a href="#cb1977-87" aria-hidden="true" tabindex="-1"></a> kableExtra           <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">4.0</span>      <span class="dv">2024-01-24</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-88"><a href="#cb1977-88" aria-hidden="true" tabindex="-1"></a> knitr                  <span class="fl">1.47</span>       <span class="dv">2024-05-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-89"><a href="#cb1977-89" aria-hidden="true" tabindex="-1"></a> labeling               <span class="dv">0</span>.<span class="fl">4.3</span>      <span class="dv">2023-08-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-90"><a href="#cb1977-90" aria-hidden="true" tabindex="-1"></a> lattice              <span class="sc">*</span> <span class="fl">0.22</span><span class="dv">-6</span>     <span class="dv">2024-03-20</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-91"><a href="#cb1977-91" aria-hidden="true" tabindex="-1"></a> lazyeval               <span class="dv">0</span>.<span class="fl">2.2</span>      <span class="dv">2019-03-15</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-92"><a href="#cb1977-92" aria-hidden="true" tabindex="-1"></a> lifecycle              <span class="dv">1</span>.<span class="fl">0.4</span>      <span class="dv">2023-11-07</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-93"><a href="#cb1977-93" aria-hidden="true" tabindex="-1"></a> locfit                 <span class="fl">1.5-9.9</span>    <span class="dv">2024-03-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-94"><a href="#cb1977-94" aria-hidden="true" tabindex="-1"></a> lubridate            <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">9.3</span>      <span class="dv">2023-09-27</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-95"><a href="#cb1977-95" aria-hidden="true" tabindex="-1"></a> magrittr             <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">0.3</span>      <span class="dv">2022-03-30</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.3</span>)</span>
<span id="cb1977-96"><a href="#cb1977-96" aria-hidden="true" tabindex="-1"></a> markdown               <span class="fl">1.13</span>       <span class="dv">2024-06-04</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-97"><a href="#cb1977-97" aria-hidden="true" tabindex="-1"></a> MASS                   <span class="fl">7.3</span><span class="dv">-61</span>     <span class="dv">2024-06-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-98"><a href="#cb1977-98" aria-hidden="true" tabindex="-1"></a> Matrix                 <span class="fl">1.7</span><span class="dv">-0</span>      <span class="dv">2024-04-26</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-99"><a href="#cb1977-99" aria-hidden="true" tabindex="-1"></a> MatrixGenerics       <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">12.3</span>     <span class="dv">2023-07-30</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-100"><a href="#cb1977-100" aria-hidden="true" tabindex="-1"></a> matrixStats          <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">3.0</span>      <span class="dv">2024-04-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-101"><a href="#cb1977-101" aria-hidden="true" tabindex="-1"></a> memoise                <span class="dv">2</span>.<span class="fl">0.1</span>      <span class="dv">2021-11-26</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-102"><a href="#cb1977-102" aria-hidden="true" tabindex="-1"></a> mgcv                   <span class="fl">1.9</span><span class="dv">-1</span>      <span class="dv">2023-12-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-103"><a href="#cb1977-103" aria-hidden="true" tabindex="-1"></a> mixsqp                 <span class="fl">0.3</span><span class="dv">-54</span>     <span class="dv">2023-12-20</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-104"><a href="#cb1977-104" aria-hidden="true" tabindex="-1"></a> multtest               <span class="dv">2</span>.<span class="fl">56.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-105"><a href="#cb1977-105" aria-hidden="true" tabindex="-1"></a> munsell                <span class="dv">0</span>.<span class="fl">5.1</span>      <span class="dv">2024-04-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-106"><a href="#cb1977-106" aria-hidden="true" tabindex="-1"></a> nlme                   <span class="fl">3.1</span><span class="dv">-165</span>    <span class="dv">2024-06-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-107"><a href="#cb1977-107" aria-hidden="true" tabindex="-1"></a> paletteer            <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">6.0</span>      <span class="dv">2024-01-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-108"><a href="#cb1977-108" aria-hidden="true" tabindex="-1"></a> patchwork            <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">2.0</span>      <span class="dv">2024-01-08</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-109"><a href="#cb1977-109" aria-hidden="true" tabindex="-1"></a> permute              <span class="sc">*</span> <span class="fl">0.9</span><span class="dv">-7</span>      <span class="dv">2022-01-27</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-110"><a href="#cb1977-110" aria-hidden="true" tabindex="-1"></a> phyloseq             <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">44.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-111"><a href="#cb1977-111" aria-hidden="true" tabindex="-1"></a> pillar                 <span class="dv">1</span>.<span class="fl">9.0</span>      <span class="dv">2023-03-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-112"><a href="#cb1977-112" aria-hidden="true" tabindex="-1"></a> pkgconfig              <span class="dv">2</span>.<span class="fl">0.3</span>      <span class="dv">2019-09-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-113"><a href="#cb1977-113" aria-hidden="true" tabindex="-1"></a> plyr                   <span class="dv">1</span>.<span class="fl">8.9</span>      <span class="dv">2023-10-02</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-114"><a href="#cb1977-114" aria-hidden="true" tabindex="-1"></a> png                    <span class="fl">0.1</span><span class="dv">-8</span>      <span class="dv">2022-11-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-115"><a href="#cb1977-115" aria-hidden="true" tabindex="-1"></a> purrr                <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">0.2</span>      <span class="dv">2023-08-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-116"><a href="#cb1977-116" aria-hidden="true" tabindex="-1"></a> R6                     <span class="dv">2</span>.<span class="fl">5.1</span>      <span class="dv">2021-08-19</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.1</span>)</span>
<span id="cb1977-117"><a href="#cb1977-117" aria-hidden="true" tabindex="-1"></a> ragg                 <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">3.2</span>      <span class="dv">2024-05-15</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-118"><a href="#cb1977-118" aria-hidden="true" tabindex="-1"></a> RColorBrewer         <span class="sc">*</span> <span class="fl">1.1</span><span class="dv">-3</span>      <span class="dv">2022-04-03</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.3</span>)</span>
<span id="cb1977-119"><a href="#cb1977-119" aria-hidden="true" tabindex="-1"></a> Rcpp                   <span class="dv">1</span>.<span class="fl">0.12</span>     <span class="dv">2024-01-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-120"><a href="#cb1977-120" aria-hidden="true" tabindex="-1"></a> RCurl                  <span class="fl">1.98-1.14</span>  <span class="dv">2024-01-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-121"><a href="#cb1977-121" aria-hidden="true" tabindex="-1"></a> readr                <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">1.5</span>      <span class="dv">2024-01-10</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-122"><a href="#cb1977-122" aria-hidden="true" tabindex="-1"></a> rematch2               <span class="dv">2</span>.<span class="fl">1.2</span>      <span class="dv">2020-05-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-123"><a href="#cb1977-123" aria-hidden="true" tabindex="-1"></a> reshape2               <span class="dv">1</span>.<span class="fl">4.4</span>      <span class="dv">2020-04-09</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-124"><a href="#cb1977-124" aria-hidden="true" tabindex="-1"></a> rhdf5                  <span class="dv">2</span>.<span class="fl">44.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-125"><a href="#cb1977-125" aria-hidden="true" tabindex="-1"></a> rhdf5filters           <span class="dv">1</span>.<span class="fl">12.1</span>     <span class="dv">2023-04-30</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-126"><a href="#cb1977-126" aria-hidden="true" tabindex="-1"></a> Rhdf5lib               <span class="dv">1</span>.<span class="fl">22.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-127"><a href="#cb1977-127" aria-hidden="true" tabindex="-1"></a> rlang                  <span class="dv">1</span>.<span class="fl">1.4</span>      <span class="dv">2024-06-04</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-128"><a href="#cb1977-128" aria-hidden="true" tabindex="-1"></a> rmarkdown              <span class="fl">2.27</span>       <span class="dv">2024-05-17</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-129"><a href="#cb1977-129" aria-hidden="true" tabindex="-1"></a> rstatix                <span class="dv">0</span>.<span class="fl">7.2</span>      <span class="dv">2023-02-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-130"><a href="#cb1977-130" aria-hidden="true" tabindex="-1"></a> rstudioapi             <span class="dv">0</span>.<span class="fl">16.0</span>     <span class="dv">2024-03-24</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-131"><a href="#cb1977-131" aria-hidden="true" tabindex="-1"></a> Rttf2pt1               <span class="dv">1</span>.<span class="fl">3.12</span>     <span class="dv">2023-01-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.2</span>)</span>
<span id="cb1977-132"><a href="#cb1977-132" aria-hidden="true" tabindex="-1"></a> S4Arrays               <span class="dv">1</span>.<span class="fl">0.6</span>      <span class="dv">2023-08-30</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-133"><a href="#cb1977-133" aria-hidden="true" tabindex="-1"></a> S4Vectors            <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">38.1</span>     <span class="dv">2023-05-02</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-134"><a href="#cb1977-134" aria-hidden="true" tabindex="-1"></a> scales               <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">3.0</span>      <span class="dv">2023-11-28</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-135"><a href="#cb1977-135" aria-hidden="true" tabindex="-1"></a> sessioninfo            <span class="dv">1</span>.<span class="fl">2.2</span>      <span class="dv">2021-12-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">1.2</span>)</span>
<span id="cb1977-136"><a href="#cb1977-136" aria-hidden="true" tabindex="-1"></a> speedyseq            <span class="sc">*</span> <span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">3.9021</span> <span class="dv">2024-06-17</span> [<span class="dv">1</span>] <span class="fu">Github</span> (mikemc<span class="sc">/</span>speedyseq<span class="sc">@</span><span class="dv">0057652</span>)</span>
<span id="cb1977-137"><a href="#cb1977-137" aria-hidden="true" tabindex="-1"></a> SQUAREM                <span class="fl">2021.1</span>     <span class="dv">2021-01-13</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">0.2</span>)</span>
<span id="cb1977-138"><a href="#cb1977-138" aria-hidden="true" tabindex="-1"></a> stringi                <span class="dv">1</span>.<span class="fl">8.4</span>      <span class="dv">2024-05-06</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-139"><a href="#cb1977-139" aria-hidden="true" tabindex="-1"></a> stringr              <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">5.1</span>      <span class="dv">2023-11-14</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-140"><a href="#cb1977-140" aria-hidden="true" tabindex="-1"></a> SummarizedExperiment <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">30.2</span>     <span class="dv">2023-06-06</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-141"><a href="#cb1977-141" aria-hidden="true" tabindex="-1"></a> survival               <span class="fl">3.7</span><span class="dv">-0</span>      <span class="dv">2024-06-05</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-142"><a href="#cb1977-142" aria-hidden="true" tabindex="-1"></a> svglite              <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">1.3</span>      <span class="dv">2023-12-08</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-143"><a href="#cb1977-143" aria-hidden="true" tabindex="-1"></a> systemfonts            <span class="dv">1</span>.<span class="fl">1.0</span>      <span class="dv">2024-05-15</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-144"><a href="#cb1977-144" aria-hidden="true" tabindex="-1"></a> textshaping            <span class="dv">0</span>.<span class="fl">4.0</span>      <span class="dv">2024-05-24</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-145"><a href="#cb1977-145" aria-hidden="true" tabindex="-1"></a> tibble               <span class="sc">*</span> <span class="dv">3</span>.<span class="fl">2.1</span>      <span class="dv">2023-03-20</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-146"><a href="#cb1977-146" aria-hidden="true" tabindex="-1"></a> tidyr                <span class="sc">*</span> <span class="dv">1</span>.<span class="fl">3.1</span>      <span class="dv">2024-01-24</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-147"><a href="#cb1977-147" aria-hidden="true" tabindex="-1"></a> tidyselect             <span class="dv">1</span>.<span class="fl">2.1</span>      <span class="dv">2024-03-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.3</span>)</span>
<span id="cb1977-148"><a href="#cb1977-148" aria-hidden="true" tabindex="-1"></a> tidytree               <span class="dv">0</span>.<span class="fl">4.6</span>      <span class="dv">2023-12-12</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-149"><a href="#cb1977-149" aria-hidden="true" tabindex="-1"></a> tidyverse            <span class="sc">*</span> <span class="dv">2</span>.<span class="fl">0.0</span>      <span class="dv">2023-02-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-150"><a href="#cb1977-150" aria-hidden="true" tabindex="-1"></a> timechange             <span class="dv">0</span>.<span class="fl">3.0</span>      <span class="dv">2024-01-18</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-151"><a href="#cb1977-151" aria-hidden="true" tabindex="-1"></a> treeio                 <span class="dv">1</span>.<span class="fl">24.3</span>     <span class="dv">2023-07-24</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-152"><a href="#cb1977-152" aria-hidden="true" tabindex="-1"></a> truncnorm              <span class="fl">1.0</span><span class="dv">-9</span>      <span class="dv">2023-03-20</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-153"><a href="#cb1977-153" aria-hidden="true" tabindex="-1"></a> tzdb                   <span class="dv">0</span>.<span class="fl">4.0</span>      <span class="dv">2023-05-12</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-154"><a href="#cb1977-154" aria-hidden="true" tabindex="-1"></a> utf8                   <span class="dv">1</span>.<span class="fl">2.4</span>      <span class="dv">2023-10-22</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.1</span>)</span>
<span id="cb1977-155"><a href="#cb1977-155" aria-hidden="true" tabindex="-1"></a> vctrs                  <span class="dv">0</span>.<span class="fl">6.5</span>      <span class="dv">2023-12-01</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-156"><a href="#cb1977-156" aria-hidden="true" tabindex="-1"></a> vegan                <span class="sc">*</span> <span class="fl">2.6-6.1</span>    <span class="dv">2024-05-21</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-157"><a href="#cb1977-157" aria-hidden="true" tabindex="-1"></a> viridis              <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">6.5</span>      <span class="dv">2024-01-29</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-158"><a href="#cb1977-158" aria-hidden="true" tabindex="-1"></a> viridisLite          <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">4.2</span>      <span class="dv">2023-05-02</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">2.3</span>)</span>
<span id="cb1977-159"><a href="#cb1977-159" aria-hidden="true" tabindex="-1"></a> visdat               <span class="sc">*</span> <span class="dv">0</span>.<span class="dv">6</span>.<span class="fl">0.9000</span> <span class="dv">2023-05-31</span> [<span class="dv">1</span>] <span class="fu">Github</span> (ropensci<span class="sc">/</span>visdat<span class="sc">@</span><span class="dv">9199906</span>)</span>
<span id="cb1977-160"><a href="#cb1977-160" aria-hidden="true" tabindex="-1"></a> withr                  <span class="dv">3</span>.<span class="fl">0.0</span>      <span class="dv">2024-01-16</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-161"><a href="#cb1977-161" aria-hidden="true" tabindex="-1"></a> xfun                   <span class="fl">0.45</span>       <span class="dv">2024-06-16</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">4.0</span>)</span>
<span id="cb1977-162"><a href="#cb1977-162" aria-hidden="true" tabindex="-1"></a> xml2                   <span class="dv">1</span>.<span class="fl">3.6</span>      <span class="dv">2023-12-04</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-163"><a href="#cb1977-163" aria-hidden="true" tabindex="-1"></a> XVector              <span class="sc">*</span> <span class="dv">0</span>.<span class="fl">40.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-164"><a href="#cb1977-164" aria-hidden="true" tabindex="-1"></a> yaml                   <span class="dv">2</span>.<span class="fl">3.8</span>      <span class="dv">2023-12-11</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-165"><a href="#cb1977-165" aria-hidden="true" tabindex="-1"></a> yulab.utils            <span class="dv">0</span>.<span class="fl">1.4</span>      <span class="dv">2024-01-28</span> [<span class="dv">1</span>] <span class="fu">CRAN</span> (R <span class="dv">4</span>.<span class="fl">3.2</span>)</span>
<span id="cb1977-166"><a href="#cb1977-166" aria-hidden="true" tabindex="-1"></a> zlibbioc               <span class="dv">1</span>.<span class="fl">46.0</span>     <span class="dv">2023-04-25</span> [<span class="dv">1</span>] Bioconductor</span>
<span id="cb1977-167"><a href="#cb1977-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1977-168"><a href="#cb1977-168" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] <span class="sc">/</span>home<span class="sc">/</span>angel<span class="sc">/</span>R<span class="sc">/</span>library</span>
<span id="cb1977-169"><a href="#cb1977-169" aria-hidden="true" tabindex="-1"></a> [<span class="dv">2</span>] <span class="sc">/</span>usr<span class="sc">/</span>local<span class="sc">/</span>lib<span class="sc">/</span>R<span class="sc">/</span>site<span class="sc">-</span>library</span>
<span id="cb1977-170"><a href="#cb1977-170" aria-hidden="true" tabindex="-1"></a> [<span class="dv">3</span>] <span class="sc">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>R<span class="sc">/</span>site<span class="sc">-</span>library</span>
<span id="cb1977-171"><a href="#cb1977-171" aria-hidden="true" tabindex="-1"></a> [<span class="dv">4</span>] <span class="sc">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>R<span class="sc">/</span>library</span>
<span id="cb1977-172"><a href="#cb1977-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1977-173"><a href="#cb1977-173" aria-hidden="true" tabindex="-1"></a>──────────────────────────────────────────────────────────────────────────────</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<p><br></p>
</div>
</div>
</section>
</section>
</section>
<section id="references" class="level2 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-love_moderated_2014" class="csl-entry" role="doc-biblioentry">
Love MI, Huber W, Anders S. <a href="https://doi.org/10.1186/s13059-014-0550-8">Moderated estimation of fold change and dispersion for <span>RNA</span>-seq data with <span>DESeq2</span></a>. <em>Genome Biol</em> 2014;<strong>15</strong>:550.
</div>
<div id="ref-youngblut_htssip_2018" class="csl-entry" role="doc-biblioentry">
Youngblut ND, Barnett SE, Buckley DH. <a href="https://doi.org/10.1371/journal.pone.0189616"><span>HTSSIP</span>: An <span>R</span> package for analysis of high throughput sequencing data from nucleic acid stable isotope probing (<span>SIP</span>) experiments</a>. <em>PLOS ONE</em> 2018;<strong>13</strong>:e0189616.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>